<!doctype html>
<html class="no-js" lang="en" data-content_root="../">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="../genindex.html" /><link rel="search" title="Search" href="../search.html" /><link rel="next" title="Toy example 1" href="toy_example1.html" /><link rel="prev" title="&lt;no title&gt;" href="../tutorials.html" />

    <!-- Generated with Sphinx 7.3.7 and Furo 2024.05.06 -->
        <title>Example of spatial multi-omics data (mouse brain with Parkinson’s disease) - MetaChat 0.0.2 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=a746c00c" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo.css?v=387cc868" />
    <link rel="stylesheet" type="text/css" href="../_static/css/sphinx_rtd_size.css?v=8f58cd29" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../_static/sg_gallery.css?v=61a4c737" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo-extensions.css?v=36a5483c" />
    <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css" />
    <link rel="stylesheet" type="text/css" href="../_static/css/override.css?v=46f5334b" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --color-brand-primary: #357473;
  --color-brand-content: #357473;
  --admonition-font-size: var(--font-size-normal);
  --admonition-title-font-size: var(--font-size-normal);
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
  <symbol id="svg-sun-with-moon" viewBox="0 0 24 24">
    <title>Auto light/dark, in light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path style="opacity: 50%" d="M 5.411 14.504 C 5.471 14.504 5.532 14.504 5.591 14.504 C 3.639 16.319 4.383 19.569 6.931 20.352 C 7.693 20.586 8.512 20.551 9.25 20.252 C 8.023 23.207 4.056 23.725 2.11 21.184 C 0.166 18.642 1.702 14.949 4.874 14.536 C 5.051 14.512 5.231 14.5 5.411 14.5 L 5.411 14.504 Z"/>
      <line x1="14.5" y1="3.25" x2="14.5" y2="1.25"/>
      <line x1="14.5" y1="15.85" x2="14.5" y2="17.85"/>
      <line x1="10.044" y1="5.094" x2="8.63" y2="3.68"/>
      <line x1="19" y1="14.05" x2="20.414" y2="15.464"/>
      <line x1="8.2" y1="9.55" x2="6.2" y2="9.55"/>
      <line x1="20.8" y1="9.55" x2="22.8" y2="9.55"/>
      <line x1="10.044" y1="14.006" x2="8.63" y2="15.42"/>
      <line x1="19" y1="5.05" x2="20.414" y2="3.636"/>
      <circle cx="14.5" cy="9.55" r="3.6"/>
    </svg>
  </symbol>
  <symbol id="svg-moon-with-sun" viewBox="0 0 24 24">
    <title>Auto light/dark, in dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path d="M 8.282 7.007 C 8.385 7.007 8.494 7.007 8.595 7.007 C 5.18 10.184 6.481 15.869 10.942 17.24 C 12.275 17.648 13.706 17.589 15 17.066 C 12.851 22.236 5.91 23.143 2.505 18.696 C -0.897 14.249 1.791 7.786 7.342 7.063 C 7.652 7.021 7.965 7 8.282 7 L 8.282 7.007 Z"/>
      <line style="opacity: 50%" x1="18" y1="3.705" x2="18" y2="2.5"/>
      <line style="opacity: 50%" x1="18" y1="11.295" x2="18" y2="12.5"/>
      <line style="opacity: 50%" x1="15.316" y1="4.816" x2="14.464" y2="3.964"/>
      <line style="opacity: 50%" x1="20.711" y1="10.212" x2="21.563" y2="11.063"/>
      <line style="opacity: 50%" x1="14.205" y1="7.5" x2="13.001" y2="7.5"/>
      <line style="opacity: 50%" x1="21.795" y1="7.5" x2="23" y2="7.5"/>
      <line style="opacity: 50%" x1="15.316" y1="10.184" x2="14.464" y2="11.036"/>
      <line style="opacity: 50%" x1="20.711" y1="4.789" x2="21.563" y2="3.937"/>
      <circle style="opacity: 50%" cx="18" cy="7.5" r="2.169"/>
    </svg>
  </symbol>
  <symbol id="svg-pencil" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-pencil-code">
      <path d="M4 20h4l10.5 -10.5a2.828 2.828 0 1 0 -4 -4l-10.5 10.5v4" />
      <path d="M13.5 6.5l4 4" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" /
    </svg>
  </symbol>
  <symbol id="svg-eye" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-eye-code">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0" />
      <path
        d="M11.11 17.958c-3.209 -.307 -5.91 -2.293 -8.11 -5.958c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6c-.21 .352 -.427 .688 -.647 1.008" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>

<a class="skip-to-content muted-link" href="#furo-main-content">Skip to content</a>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../index.html"><div class="brand">MetaChat 0.0.2 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
          <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo" src="../_static/logo.png" alt="Logo"/>
  </div>
  
  
</a><form class="sidebar-search-container" method="get" action="../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">Example of spatial multi-omics data (mouse brain with Parkinson’s disease)</a></li>
<li class="toctree-l1"><a class="reference internal" href="toy_example1.html">Toy example 1</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../api/index.html">API</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle navigation of API</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pp.MetaChatDB.html">metachat.pp.MetaChatDB</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pp.LRC_unfiltered.html">metachat.pp.LRC_unfiltered</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pp.LRC_cluster.html">metachat.pp.LRC_cluster</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pp.LRC_filtered.html">metachat.pp.LRC_filtered</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pp.compute_costDistance.html">metachat.pp.compute_costDistance</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.metabolic_communication.html">metachat.tl.metabolic_communication</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.summary_communication.html">metachat.tl.summary_communication</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_flow.html">metachat.tl.communication_flow</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_group.html">metachat.tl.communication_group</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_group_spatial.html">metachat.tl.communication_group_spatial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.summary_pathway.html">metachat.tl.summary_pathway</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_responseGenes.html">metachat.tl.communication_responseGenes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_responseGenes_cluster.html">metachat.tl.communication_responseGenes_cluster</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_responseGenes_keggEnrich.html">metachat.tl.communication_responseGenes_keggEnrich</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_communication_flow.html">metachat.pl.plot_communication_flow</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_group_communication_chord.html">metachat.pl.plot_group_communication_chord</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_group_communication_heatmap.html">metachat.pl.plot_group_communication_heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_communication_responseGenes.html">metachat.pl.plot_communication_responseGenes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_group_communication_compare_hierarchy_diagram.html">metachat.pl.plot_group_communication_compare_hierarchy_diagram</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_MSpair_contribute_group.html">metachat.pl.plot_MSpair_contribute_group</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_communication_responseGenes_keggEnrich.html">metachat.pl.plot_communication_responseGenes_keggEnrich</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_summary_pathway.html">metachat.pl.plot_summary_pathway</a></li>
</ul>
</li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          

<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
              <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main" id="furo-main-content">
          <section id="example-of-spatial-multi-omics-data-mouse-brain-with-parkinson-s-disease">
<h1>Example of spatial multi-omics data (mouse brain with Parkinson’s disease)<a class="headerlink" href="#example-of-spatial-multi-omics-data-mouse-brain-with-parkinson-s-disease" title="Link to this heading"></a></h1>
<p>In this tutorial, we use mouse brain slices as an example to show the basic application of MetaChat to multi-omics data of transcriptomics and metabolomics [<a class="reference external" href="https://www.nature.com/articles/s41587-023-01937-y">Marco et al., 2023</a>].</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Importing packages</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">scanpy</span> <span class="k">as</span> <span class="nn">sc</span>
<span class="kn">import</span> <span class="nn">squidpy</span> <span class="k">as</span> <span class="nn">sq</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>

<span class="kn">import</span> <span class="nn">metachat</span> <span class="k">as</span> <span class="nn">mc</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Setting your work dictionary</span>
<span class="n">os</span><span class="o">.</span><span class="n">chdir</span><span class="p">(</span><span class="s2">&quot;/home/Project/MetaChat_test/mouse_brain&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<section id="data-preprocessing">
<h2>Data preprocessing<a class="headerlink" href="#data-preprocessing" title="Link to this heading"></a></h2>
<section id="preprocessing-for-transcriptomics-data">
<h3>Preprocessing for transcriptomics data<a class="headerlink" href="#preprocessing-for-transcriptomics-data" title="Link to this heading"></a></h3>
<p>First, we perform basic preprocessing of the transcriptome, including data transformation, highly variable genes, clustering, and reduction visualization. All processes are handled by Scanpy [<a class="reference external" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1382-0">F. Alexander et al., 2018</a>] and Squidpy [<a class="reference external" href="https://www.nature.com/articles/s41592-021-01358-2">Giovanni et al., 2022</a>].</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">adata_RNA</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="s1">&#39;data/multiomics_mousebrain_parkinson_RNA.h5ad&#39;</span><span class="p">)</span>
<span class="n">adata_RNA</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AnnData object with n_obs × n_vars = 3011 × 32285
    obs: &#39;in_tissue&#39;, &#39;array_row&#39;, &#39;array_col&#39;
    var: &#39;gene_ids&#39;, &#39;feature_types&#39;, &#39;genome&#39;
    uns: &#39;spatial&#39;
    obsm: &#39;spatial&#39;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">adata_RNA</span><span class="o">.</span><span class="n">var_names_make_unique</span><span class="p">()</span>
<span class="n">adata_RNA</span><span class="o">.</span><span class="n">raw</span> <span class="o">=</span> <span class="n">adata_RNA</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">normalize_total</span><span class="p">(</span><span class="n">adata_RNA</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata_RNA</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">adata_visual</span> <span class="o">=</span> <span class="n">adata_RNA</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">random_seed</span> <span class="o">=</span> <span class="mi">42</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">highly_variable_genes</span><span class="p">(</span><span class="n">adata_visual</span><span class="p">,</span> <span class="n">min_mean</span><span class="o">=</span><span class="mf">0.0125</span><span class="p">,</span> <span class="n">max_mean</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">min_disp</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>
<span class="n">adata_visual</span> <span class="o">=</span> <span class="n">adata_visual</span><span class="p">[:,</span> <span class="n">adata_visual</span><span class="o">.</span><span class="n">var</span><span class="o">.</span><span class="n">highly_variable</span><span class="p">]</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">pca</span><span class="p">(</span><span class="n">adata_visual</span><span class="p">,</span> <span class="n">svd_solver</span><span class="o">=</span><span class="s1">&#39;arpack&#39;</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="n">random_seed</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">neighbors</span><span class="p">(</span><span class="n">adata_visual</span><span class="p">,</span> <span class="n">n_neighbors</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">n_pcs</span><span class="o">=</span><span class="mi">40</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="n">random_seed</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">umap</span><span class="p">(</span><span class="n">adata_visual</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="n">random_seed</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">leiden</span><span class="p">(</span><span class="n">adata_visual</span><span class="p">,</span> <span class="n">resolution</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">spatial</span><span class="p">(</span><span class="n">adata_visual</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;leiden&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/01aa2c6e1a77298c6f568ea28caf7e305a63b7872932bf542332493ce95782af.png" src="../_images/01aa2c6e1a77298c6f568ea28caf7e305a63b7872932bf542332493ce95782af.png" />
</div>
</div>
<p>By comparing with <a class="reference external" href="https://mouse.brain-map.org/experiment/thumbnails/100048576?image_type=atlas">Allen Mouse Brain Atlas</a> [<a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0092867420304025">Quanxin et al., 2020</a>], we can initially get the following six basic types of organizations.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">cluster_to_cell_type</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;0&#39;</span><span class="p">:</span> <span class="s1">&#39;Striatum dorsal region&#39;</span><span class="p">,</span>
    <span class="s1">&#39;1&#39;</span><span class="p">:</span> <span class="s1">&#39;Cerebral cortex&#39;</span><span class="p">,</span>
    <span class="s1">&#39;2&#39;</span><span class="p">:</span> <span class="s1">&#39;Striatum ventral region&#39;</span><span class="p">,</span>
    <span class="s1">&#39;3&#39;</span><span class="p">:</span> <span class="s1">&#39;Cerebral cortex&#39;</span><span class="p">,</span>
    <span class="s1">&#39;4&#39;</span><span class="p">:</span> <span class="s1">&#39;Lateral septal complex&#39;</span><span class="p">,</span>
    <span class="s1">&#39;5&#39;</span><span class="p">:</span> <span class="s1">&#39;Fiber tracts&#39;</span><span class="p">,</span>
    <span class="s1">&#39;6&#39;</span><span class="p">:</span> <span class="s1">&#39;Cerebral cortex&#39;</span><span class="p">,</span>
    <span class="s1">&#39;7&#39;</span><span class="p">:</span> <span class="s1">&#39;Fiber tracts&#39;</span><span class="p">,</span>
    <span class="s1">&#39;8&#39;</span><span class="p">:</span> <span class="s1">&#39;Cerebral cortex&#39;</span><span class="p">,</span>
    <span class="s1">&#39;9&#39;</span><span class="p">:</span> <span class="s1">&#39;Cerebral cortex&#39;</span><span class="p">,</span>
    <span class="s1">&#39;10&#39;</span><span class="p">:</span> <span class="s1">&#39;Striatum dorsal region&#39;</span><span class="p">,</span>
    <span class="s1">&#39;11&#39;</span><span class="p">:</span> <span class="s1">&#39;Striatum ventral region&#39;</span><span class="p">,</span>
    <span class="s1">&#39;12&#39;</span><span class="p">:</span> <span class="s1">&#39;Ventricular systems&#39;</span><span class="p">,</span>
    <span class="s1">&#39;13&#39;</span><span class="p">:</span> <span class="s1">&#39;Cerebral cortex&#39;</span><span class="p">,</span>
<span class="p">}</span>
<span class="n">adata_RNA</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;tissue_type&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata_visual</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;leiden&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="n">cluster_to_cell_type</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span> <span class="o">=</span> <span class="p">(</span><span class="mi">5</span><span class="p">,</span><span class="mi">5</span><span class="p">))</span>
<span class="n">sq</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">spatial_scatter</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata_RNA</span><span class="p">,</span>
                      <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;tissue_type&quot;</span><span class="p">,</span>
                      <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_box_aspect</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/87cb0cd46421743378e882cda4b90dfa336b97d93caf05557c4ed1eee73f30f3.png" src="../_images/87cb0cd46421743378e882cda4b90dfa336b97d93caf05557c4ed1eee73f30f3.png" />
</div>
</div>
</section>
<section id="preprocessing-for-metabolomics-data">
<h3>Preprocessing for metabolomics data<a class="headerlink" href="#preprocessing-for-metabolomics-data" title="Link to this heading"></a></h3>
<p>Next, metabolomic data is imported and spatial information from the transcriptomic data is assigned to the metabolome.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">adata_MSI</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="s1">&#39;data/multiomics_mousebrain_parkinson_MSI.h5ad&#39;</span><span class="p">)</span>
<span class="n">adata_MSI</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata_RNA</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span>
<span class="n">adata_MSI</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata_RNA</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span>
<span class="n">adata_MSI</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AnnData object with n_obs × n_vars = 3011 × 3015
    obs: &#39;orig.ident&#39;, &#39;nCount_RNA&#39;, &#39;nFeature_RNA&#39;
    var: &#39;features&#39;
    uns: &#39;spatial&#39;
    obsm: &#39;spatial&#39;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">adata_MSI</span><span class="o">.</span><span class="n">var_names_make_unique</span><span class="p">()</span>
<span class="n">adata_MSI</span><span class="o">.</span><span class="n">raw</span> <span class="o">=</span> <span class="n">adata_MSI</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">normalize_total</span><span class="p">(</span><span class="n">adata_MSI</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The MSI annotations for the metabolome should have the HMDB ID from <a class="reference external" href="https://hmdb.ca/">The Human Metabolome Database</a>  as the variable name in the adata object, due to the fact that the built-in database MetaChatDB is indexed by the HMDB ID.</p>
</div>
</section>
<section id="combine-the-transcriptomics-and-metabolomics-data">
<h3>Combine the transcriptomics and metabolomics data<a class="headerlink" href="#combine-the-transcriptomics-and-metabolomics-data" title="Link to this heading"></a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">scipy.sparse</span> <span class="kn">import</span> <span class="n">hstack</span>
<span class="n">X_combined</span> <span class="o">=</span> <span class="n">hstack</span><span class="p">([</span><span class="n">adata_RNA</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">adata_MSI</span><span class="o">.</span><span class="n">X</span><span class="p">])</span>
<span class="n">X_combined</span> <span class="o">=</span> <span class="n">X_combined</span><span class="o">.</span><span class="n">tocsr</span><span class="p">()</span>
<span class="n">var_combined</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s2">&quot;features&quot;</span> <span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">concatenate</span><span class="p">((</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">adata_RNA</span><span class="o">.</span><span class="n">var_names</span><span class="p">),</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">adata_MSI</span><span class="o">.</span><span class="n">var_names</span><span class="p">)))})</span>
<span class="n">var_combined</span><span class="o">.</span><span class="n">index</span> <span class="o">=</span> <span class="n">var_combined</span><span class="p">[</span><span class="s2">&quot;features&quot;</span><span class="p">]</span>
<span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">AnnData</span><span class="p">(</span><span class="n">X_combined</span><span class="p">,</span> <span class="n">obs</span><span class="o">=</span><span class="n">adata_RNA</span><span class="o">.</span><span class="n">obs</span><span class="p">,</span> <span class="n">var</span><span class="o">=</span><span class="n">var_combined</span><span class="p">)</span>
<span class="n">adata</span><span class="o">.</span><span class="n">obsm</span> <span class="o">=</span> <span class="n">adata_RNA</span><span class="o">.</span><span class="n">obsm</span>
<span class="n">adata</span><span class="o">.</span><span class="n">uns</span> <span class="o">=</span> <span class="n">adata_RNA</span><span class="o">.</span><span class="n">uns</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="mcc-inference">
<h2>MCC Inference<a class="headerlink" href="#mcc-inference" title="Link to this heading"></a></h2>
<section id="import-metachatdb">
<h3>Import MetaChatDB<a class="headerlink" href="#import-metachatdb" title="Link to this heading"></a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">df_metasen</span> <span class="o">=</span> <span class="n">mc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">MetaChatDB</span><span class="p">(</span><span class="n">species</span><span class="o">=</span><span class="s1">&#39;mouse&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>If you have your own metabolic-sensor database, follow the tutorial How to Build a Custom Database.</p>
</div>
</section>
<section id="identify-long-range-channels-lrc">
<h3>Identify long-range channels (LRC)<a class="headerlink" href="#identify-long-range-channels-lrc" title="Link to this heading"></a></h3>
<p>Next, we will see how to identify LRCs. There are two ways to do this: (i) by marker genes, as here we utilize the high expression of Mgp as a potential spatial location for blood vessels or Gfap for cerebrospinal fluid (CSF); (ii) Manual determination of LRC through user-defined regions.</p>
<p>Here we show the process of identifying LRC of cerebrospinal fluid.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">LRC_name</span> <span class="o">=</span> <span class="s2">&quot;CSF&quot;</span>
<span class="n">LRC_marker_gene</span> <span class="o">=</span> <span class="s2">&quot;Gfap&quot;</span>

<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span> <span class="o">=</span> <span class="p">(</span><span class="mi">6</span><span class="p">,</span><span class="mi">5</span><span class="p">))</span>
<span class="n">sq</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">spatial_scatter</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span>
                      <span class="n">color</span> <span class="o">=</span> <span class="p">[</span><span class="n">LRC_marker_gene</span><span class="p">],</span>
                      <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_box_aspect</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/8926c12781603d61f2b7b3cd44eff75e52c7658834ff77f7e7eb8157c4e4ad7e.png" src="../_images/8926c12781603d61f2b7b3cd44eff75e52c7658834ff77f7e7eb8157c4e4ad7e.png" />
</div>
</div>
<p>We can see a clear channel near the fiber tracts. Next, we select the spots with expression above the 85% quantile as candidate spots for LRC.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">LRC_unfiltered</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                     <span class="n">LRC_name</span> <span class="o">=</span> <span class="n">LRC_name</span><span class="p">,</span>
                     <span class="n">LRC_marker_gene</span> <span class="o">=</span> <span class="n">LRC_marker_gene</span><span class="p">,</span>
                     <span class="n">quantile</span> <span class="o">=</span> <span class="mi">85</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Cells with Gfap expression above the 85% have been selected as candidates and stored in &#39;adata.obs[&#39;LRC_CSF_unfiltered&#39;]&#39;.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span> <span class="o">=</span> <span class="p">(</span><span class="mi">6</span><span class="p">,</span><span class="mi">5</span><span class="p">))</span>
<span class="n">sq</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">spatial_scatter</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span>
                      <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;LRC_&#39;</span> <span class="o">+</span> <span class="n">LRC_name</span> <span class="o">+</span> <span class="s1">&#39;_unfiltered&#39;</span><span class="p">,</span>
                      <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_box_aspect</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/ffa0ccaa97f333c7f330eac7fcd18ed1da2ff1f7b62851e4fa5b6d0b0f3a618e.png" src="../_images/ffa0ccaa97f333c7f330eac7fcd18ed1da2ff1f7b62851e4fa5b6d0b0f3a618e.png" />
</div>
</div>
<p>We can see that the candidate spots have a clear channel, but are accompanied by scattered dots. We identify non-elliptical LRC clusters and exclude outliers by a local density clustering algorithm. Of these, points that satisfy both <code class="docutils literal notranslate"><span class="pre">density</span> <span class="pre">&gt;</span> <span class="pre">density_cutoff</span></code> and <code class="docutils literal notranslate"><span class="pre">delta</span> <span class="pre">&gt;</span> <span class="pre">delta_cutoff</span></code> will be used as cluster centers (red solid line). And points with <code class="docutils literal notranslate"><span class="pre">density</span> <span class="pre">&lt;</span> <span class="pre">outlier_cutoff</span></code> will be considered as outliers spots.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">LRC_cluster_CSF</span> <span class="o">=</span> <span class="n">mc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">LRC_cluster</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                                    <span class="n">LRC_name</span> <span class="o">=</span> <span class="n">LRC_name</span><span class="p">,</span>
                                    <span class="n">density_cutoff</span> <span class="o">=</span> <span class="mi">7</span><span class="p">,</span>
                                    <span class="n">delta_cutoff</span> <span class="o">=</span> <span class="mi">8000</span><span class="p">,</span>
                                    <span class="n">outlier_cutoff</span> <span class="o">=</span> <span class="mf">4.5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/92b06efaff4b8da010d4289173e2338126a354cab5d4e71a12b6b1b1edba20ba.png" src="../_images/92b06efaff4b8da010d4289173e2338126a354cab5d4e71a12b6b1b1edba20ba.png" />
</div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>This step allows you to keep adjusting the parameters (<code class="docutils literal notranslate"><span class="pre">density_cutoff</span></code>, <code class="docutils literal notranslate"><span class="pre">delta_cutoff</span></code> and <code class="docutils literal notranslate"><span class="pre">outlier_cutoff</span></code>) to get the best visual effect.</p>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">LRC_filtered</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                   <span class="n">LRC_name</span> <span class="o">=</span> <span class="n">LRC_name</span><span class="p">,</span>
                   <span class="n">LRC_cluster</span> <span class="o">=</span> <span class="n">LRC_cluster_CSF</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Candidate points for CSF LRC are clustered and outliers are removed. LRC points are stored in &#39;adata.obs[&#39;LRC_CSF_filtered&#39;]&#39;.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">matplotlib.colors</span> <span class="k">as</span> <span class="nn">mcolors</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span> <span class="o">=</span> <span class="p">(</span><span class="mi">6</span><span class="p">,</span><span class="mi">5</span><span class="p">))</span>
<span class="n">sq</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">spatial_scatter</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span>
                      <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;LRC_&#39;</span> <span class="o">+</span> <span class="n">LRC_name</span> <span class="o">+</span> <span class="s1">&#39;_filtered&#39;</span><span class="p">,</span>
                      <span class="n">alpha</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span>
                      <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_box_aspect</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/b6b77989f732cf74dc670aee46fb8fac484ef1f368d302c16d497a81f1bc7a30.png" src="../_images/b6b77989f732cf74dc670aee46fb8fac484ef1f368d302c16d497a81f1bc7a30.png" />
</div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>If you want to use manual determination of the LRC, please see the tutorial for details.</p>
</div>
<p>Next, we want to compute the cost distance for this data, based on the previously identified LRCs. for the case of multiple LRC types, enter all of them together in a list, e.g. LRC_type = [“Blood”, “CSF”].</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Since the coordinate scale may vary for each data, it is difficult to intuitively estimate the parameter <code class="docutils literal notranslate"><span class="pre">dis_thr</span></code>, which reflects the furthest distance of influence by LRC. if you set <code class="docutils literal notranslate"><span class="pre">plot=True</span></code>, you can clearly see for each class of each type, which spots may be affected by the LRC, and thus determine <code class="docutils literal notranslate"><span class="pre">dis_thr</span></code>.</p>
<p>For <code class="docutils literal notranslate"><span class="pre">LRC_strength</span></code>, this is generally an order of magnitude greater than the minimum distance between spots.</p>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">compute_costDistance</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                           <span class="n">LRC_type</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;CSF&quot;</span><span class="p">],</span>
                           <span class="n">dis_thr</span> <span class="o">=</span> <span class="mi">1000</span><span class="p">,</span>
                           <span class="n">k_neighb</span> <span class="o">=</span> <span class="mi">5</span><span class="p">,</span>
                           <span class="n">LRC_strength</span> <span class="o">=</span> <span class="mi">2000</span><span class="p">,</span>
                           <span class="n">plot</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Compute spatial cost distance without long-range channel...
Compute spatial cost distance incorporating long-range channel of CSF
The points that can occur long-range communication is...
</pre></div>
</div>
<img alt="../_images/d2908b3796043ff7a293a4d684638a256f00e8d8484b72b563c95028a43d48cc.png" src="../_images/d2908b3796043ff7a293a4d684638a256f00e8d8484b72b563c95028a43d48cc.png" />
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>The points that can occur long-range communication is...
</pre></div>
</div>
<img alt="../_images/cce75b3e9691126f513c3cef5fe88f90c1de1dc32ed67d9983ee1b851255e2c6.png" src="../_images/cce75b3e9691126f513c3cef5fe88f90c1de1dc32ed67d9983ee1b851255e2c6.png" />
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>  Construct network graph of long-range channel among 5 neighborhoods...
  Calculate the shortest path distance from the source to the target using the shortest path algorithm...
    For the long-range case of cluster 1...
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 190/190 [00:14&lt;00:00, 13.05it/s]
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>    For the long-range case of cluster 2...
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 14/14 [00:00&lt;00:00, 1925.51it/s]
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>  Rearrange distance matrix...
    For the long-range case of cluster 1...
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 448/448 [00:00&lt;00:00, 676.06it/s]
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>    For the long-range case of cluster 2...
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 36/36 [00:00&lt;00:00, 8275.06it/s]
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Finished!
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
</div>
<p>With the previous preparation, we can finally begin to make inferences about metabolic cellular communication.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>It is better to have the same <code class="docutils literal notranslate"><span class="pre">dis_thr</span></code> of this function as in <code class="docutils literal notranslate"><span class="pre">dis_thr</span></code> in <code class="docutils literal notranslate"><span class="pre">mc.pp.compute_costDistance</span></code>.</p>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">metabolic_communication</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                              <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                              <span class="n">df_metasen</span> <span class="o">=</span> <span class="n">df_metasen</span><span class="p">,</span>
                              <span class="n">LRC_type</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;CSF&quot;</span><span class="p">],</span>
                              <span class="n">dis_thr</span> <span class="o">=</span> <span class="mi">1000</span><span class="p">)</span>
<span class="n">adata</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;results/inference_results.h5ad&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>There are 327 pairs were found from the spatial data.
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="downstream-analysis">
<h2>Downstream analysis<a class="headerlink" href="#downstream-analysis" title="Link to this heading"></a></h2>
<p>Next, based on our previously obtained inferences, we perform a series of downstream analyses and visualizations.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s2">&quot;results/inference_results.h5ad&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<section id="summary-communication-among-different-scales">
<h3>Summary communication among different scales<a class="headerlink" href="#summary-communication-among-different-scales" title="Link to this heading"></a></h3>
<p>Since we inferred the MCC signals for each metabolite-sensor pair, and we often have a need to analyze a particular metabolite and a particular metabolic pathway or custom signal set, we need to summarize the signal sets at different scales in advance.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">sum_metabolites</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;HMDB0000216&#39;</span><span class="p">,</span><span class="s1">&#39;HMDB0000073&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">summary_communication</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                            <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                            <span class="n">sum_metabolites</span> <span class="o">=</span> <span class="n">sum_metabolites</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This function supports multiple inputs summarizing signals at the same time, for example
mc.tl.summary_communication(adata = adata,
database_name = ‘MetaChatDB’,
sum_metabolites = [‘HMDB0000216’,’HMDB0000073’],
sum_metabolites = [‘HMDB0000216’,’HMDB0000073’],
sum_metabolites = [‘HMDB0000216’,’HMDB0000073’])</p>
</div>
</section>
<section id="identifing-mcc-flow">
<h3>Identifing MCC flow<a class="headerlink" href="#identifing-mcc-flow" title="Link to this heading"></a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">communication_flow</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                         <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                         <span class="n">sum_metabolites</span> <span class="o">=</span> <span class="n">sum_metabolites</span><span class="p">,</span>
                         <span class="n">k</span> <span class="o">=</span> <span class="mi">5</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">15</span><span class="p">,</span><span class="mi">5</span><span class="p">))</span>
<span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_communication_flow</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span> 
                              <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                              <span class="n">metabolite_name</span> <span class="o">=</span> <span class="s1">&#39;HMDB0000073&#39;</span><span class="p">,</span>
                              <span class="n">plot_method</span> <span class="o">=</span> <span class="s1">&#39;grid&#39;</span><span class="p">,</span>
                              <span class="n">background</span> <span class="o">=</span> <span class="s1">&#39;image&#39;</span><span class="p">,</span>
                              <span class="n">group_name</span> <span class="o">=</span> <span class="s1">&#39;tissue_type&#39;</span><span class="p">,</span>
                              <span class="n">summary</span> <span class="o">=</span> <span class="s1">&#39;sender&#39;</span><span class="p">,</span>
                              <span class="n">ndsize</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
                              <span class="n">scale</span> <span class="o">=</span> <span class="mf">0.0000000000003</span><span class="p">,</span>
                              <span class="n">normalize_v_quantile</span> <span class="o">=</span> <span class="mf">0.995</span><span class="p">,</span>
                              <span class="n">grid_density</span> <span class="o">=</span> <span class="mf">0.4</span><span class="p">,</span>
                              <span class="n">title</span> <span class="o">=</span> <span class="s1">&#39;Image&#39;</span><span class="p">,</span>   
                              <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">ax</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_box_aspect</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>

<span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_communication_flow</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span> 
                              <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                              <span class="n">metabolite_name</span> <span class="o">=</span> <span class="s1">&#39;HMDB0000073&#39;</span><span class="p">,</span>
                              <span class="n">plot_method</span> <span class="o">=</span> <span class="s1">&#39;grid&#39;</span><span class="p">,</span>
                              <span class="n">background</span> <span class="o">=</span> <span class="s1">&#39;group&#39;</span><span class="p">,</span>
                              <span class="n">group_name</span> <span class="o">=</span> <span class="s1">&#39;tissue_type&#39;</span><span class="p">,</span>
                              <span class="n">summary</span> <span class="o">=</span> <span class="s1">&#39;sender&#39;</span><span class="p">,</span>
                              <span class="n">ndsize</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
                              <span class="n">scale</span> <span class="o">=</span> <span class="mf">0.0000000000003</span><span class="p">,</span>
                              <span class="n">normalize_v_quantile</span> <span class="o">=</span> <span class="mf">0.995</span><span class="p">,</span>
                              <span class="n">grid_density</span> <span class="o">=</span> <span class="mf">0.4</span><span class="p">,</span>
                              <span class="n">title</span> <span class="o">=</span> <span class="s1">&#39;Group types&#39;</span><span class="p">,</span>   
                              <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
<span class="n">ax</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_box_aspect</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>

<span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_communication_flow</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span> 
                              <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                              <span class="n">metabolite_name</span> <span class="o">=</span> <span class="s1">&#39;HMDB0000073&#39;</span><span class="p">,</span>
                              <span class="n">plot_method</span> <span class="o">=</span> <span class="s1">&#39;grid&#39;</span><span class="p">,</span>
                              <span class="n">background</span> <span class="o">=</span> <span class="s1">&#39;summary&#39;</span><span class="p">,</span>
                              <span class="n">group_name</span> <span class="o">=</span> <span class="s1">&#39;tissue_type&#39;</span><span class="p">,</span>
                              <span class="n">summary</span> <span class="o">=</span> <span class="s1">&#39;sender&#39;</span><span class="p">,</span>
                              <span class="n">cmap</span> <span class="o">=</span> <span class="s1">&#39;YlOrRd&#39;</span><span class="p">,</span>
                              <span class="n">ndsize</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
                              <span class="n">scale</span> <span class="o">=</span> <span class="mf">0.0000000000003</span><span class="p">,</span>
                              <span class="n">normalize_v_quantile</span> <span class="o">=</span> <span class="mf">0.995</span><span class="p">,</span>
                              <span class="n">grid_density</span> <span class="o">=</span> <span class="mf">0.4</span><span class="p">,</span>
                              <span class="n">title</span> <span class="o">=</span> <span class="s1">&#39;MCC Signals&#39;</span><span class="p">,</span>   
                              <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="mi">2</span><span class="p">])</span>
<span class="n">ax</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">.</span><span class="n">set_box_aspect</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/38203002ddce856486407052cc255ec9941a2067304af17a90e6598d06df3fbe.png" src="../_images/38203002ddce856486407052cc255ec9941a2067304af17a90e6598d06df3fbe.png" />
</div>
</div>
</section>
<section id="constructing-group-level-mcc">
<h3>Constructing group-level MCC<a class="headerlink" href="#constructing-group-level-mcc" title="Link to this heading"></a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">communication_group</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                          <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                          <span class="n">group_name</span> <span class="o">=</span> <span class="s2">&quot;tissue_type&quot;</span><span class="p">,</span>
                          <span class="n">sum_metabolites</span> <span class="o">=</span> <span class="n">sum_metabolites</span><span class="p">,</span>
                          <span class="n">n_permutations</span> <span class="o">=</span> <span class="mi">100</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">adata</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;MetaChat_group-tissue_type-MetaChatDB-HMDB0000073&#39;</span><span class="p">][</span><span class="s1">&#39;communication_matrix&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Cerebral cortex</th>
      <th>Fiber tracts</th>
      <th>Lateral septal complex</th>
      <th>Striatum dorsal region</th>
      <th>Striatum ventral region</th>
      <th>Ventricular systems</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Cerebral cortex</th>
      <td>1.060052e-11</td>
      <td>5.985997e-12</td>
      <td>5.367747e-12</td>
      <td>3.851556e-12</td>
      <td>5.980569e-12</td>
      <td>1.108535e-11</td>
    </tr>
    <tr>
      <th>Fiber tracts</th>
      <td>8.941187e-12</td>
      <td>1.756259e-11</td>
      <td>1.813045e-17</td>
      <td>1.101182e-11</td>
      <td>1.427381e-11</td>
      <td>2.586341e-11</td>
    </tr>
    <tr>
      <th>Lateral septal complex</th>
      <td>1.827025e-12</td>
      <td>5.856315e-13</td>
      <td>1.715571e-11</td>
      <td>3.597803e-13</td>
      <td>3.308491e-13</td>
      <td>1.257162e-12</td>
    </tr>
    <tr>
      <th>Striatum dorsal region</th>
      <td>5.726325e-12</td>
      <td>1.436869e-11</td>
      <td>8.287509e-18</td>
      <td>3.606240e-10</td>
      <td>3.068785e-11</td>
      <td>1.409258e-11</td>
    </tr>
    <tr>
      <th>Striatum ventral region</th>
      <td>2.451099e-12</td>
      <td>3.763631e-12</td>
      <td>3.598783e-18</td>
      <td>1.658457e-11</td>
      <td>2.576112e-10</td>
      <td>3.478752e-12</td>
    </tr>
    <tr>
      <th>Ventricular systems</th>
      <td>7.851767e-12</td>
      <td>2.412584e-11</td>
      <td>1.282194e-17</td>
      <td>9.945576e-12</td>
      <td>1.285908e-11</td>
      <td>4.135405e-10</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">adata</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;MetaChat_group-tissue_type-MetaChatDB-HMDB0000073&#39;</span><span class="p">][</span><span class="s1">&#39;communication_pvalue&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Cerebral cortex</th>
      <th>Fiber tracts</th>
      <th>Lateral septal complex</th>
      <th>Striatum dorsal region</th>
      <th>Striatum ventral region</th>
      <th>Ventricular systems</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Cerebral cortex</th>
      <td>1.00</td>
      <td>1.00</td>
      <td>0.99</td>
      <td>1.00</td>
      <td>0.99</td>
      <td>0.37</td>
    </tr>
    <tr>
      <th>Fiber tracts</th>
      <td>0.93</td>
      <td>1.00</td>
      <td>1.00</td>
      <td>0.48</td>
      <td>0.18</td>
      <td>0.13</td>
    </tr>
    <tr>
      <th>Lateral septal complex</th>
      <td>1.00</td>
      <td>1.00</td>
      <td>1.00</td>
      <td>1.00</td>
      <td>1.00</td>
      <td>0.78</td>
    </tr>
    <tr>
      <th>Striatum dorsal region</th>
      <td>1.00</td>
      <td>0.18</td>
      <td>1.00</td>
      <td>0.00</td>
      <td>0.00</td>
      <td>0.26</td>
    </tr>
    <tr>
      <th>Striatum ventral region</th>
      <td>1.00</td>
      <td>1.00</td>
      <td>1.00</td>
      <td>0.01</td>
      <td>0.00</td>
      <td>0.73</td>
    </tr>
    <tr>
      <th>Ventricular systems</th>
      <td>0.72</td>
      <td>0.13</td>
      <td>1.00</td>
      <td>0.40</td>
      <td>0.26</td>
      <td>0.97</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">communication_group_spatial</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                                  <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                                  <span class="n">group_name</span> <span class="o">=</span> <span class="s2">&quot;tissue_type&quot;</span><span class="p">,</span>
                                  <span class="n">sum_metabolites</span> <span class="o">=</span> <span class="n">sum_metabolites</span><span class="p">,</span>
                                  <span class="n">n_permutations</span> <span class="o">=</span> <span class="mi">100</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Computing group-level MCC for MetaChatDB-HMDB0000216
Computing group-level MCC for MetaChatDB-HMDB0000073
Computing group-level MCC for MetaChatDB-total-total
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Processing pairs: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 36/36 [08:56&lt;00:00, 14.90s/it]
</pre></div>
</div>
</div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Let’s give readers a helpful hint!</p>
</div>
<p>Correspondingly, the group-level MCC results considering the spatial distance distribution are stored in <code class="docutils literal notranslate"><span class="pre">adata.uns['MetaChat_group_spatial-tissue_type-MetaChatDB-HMDB0000073']</span></code></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">11</span><span class="p">,</span><span class="mf">4.5</span><span class="p">),</span> <span class="n">subplot_kw</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;projection&#39;</span><span class="p">:</span> <span class="s1">&#39;polar&#39;</span><span class="p">})</span>
<span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_group_communication_chord</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                                     <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                                     <span class="n">group_name</span> <span class="o">=</span> <span class="s2">&quot;tissue_type&quot;</span><span class="p">,</span>
                                     <span class="n">metabolite_name</span> <span class="o">=</span> <span class="s1">&#39;HMDB0000073&#39;</span><span class="p">,</span>
                                     <span class="n">permutation_spatial</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                                     <span class="n">p_value_cutoff</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span>
                                     <span class="n">space</span> <span class="o">=</span> <span class="mi">10</span><span class="p">,</span>
                                     <span class="n">self_communication_off</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                                     <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">ax</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;With Self Communication&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>

<span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_group_communication_chord</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                                     <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                                     <span class="n">group_name</span> <span class="o">=</span> <span class="s2">&quot;tissue_type&quot;</span><span class="p">,</span>
                                     <span class="n">metabolite_name</span> <span class="o">=</span> <span class="s1">&#39;HMDB0000073&#39;</span><span class="p">,</span>
                                     <span class="n">permutation_spatial</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                                     <span class="n">p_value_cutoff</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span>
                                     <span class="n">space</span> <span class="o">=</span> <span class="mi">10</span><span class="p">,</span>
                                     <span class="n">self_communication_off</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                                     <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
<span class="n">ax</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;Without Self Communication&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Text(0.5, 1.0, &#39;Without Self Communication&#39;)
</pre></div>
</div>
<img alt="../_images/2c20dea46391bf6541e26b3927e68fd5faf4e6b7cab735e9f35981080bdd46d9.png" src="../_images/2c20dea46391bf6541e26b3927e68fd5faf4e6b7cab735e9f35981080bdd46d9.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">11</span><span class="p">,</span><span class="mi">3</span><span class="p">))</span>
<span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_group_communication_heatmap</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                                       <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                                       <span class="n">group_name</span> <span class="o">=</span> <span class="s2">&quot;tissue_type&quot;</span><span class="p">,</span>
                                       <span class="n">metabolite_name</span> <span class="o">=</span> <span class="s1">&#39;HMDB0000073&#39;</span><span class="p">,</span>
                                       <span class="n">cmap</span> <span class="o">=</span> <span class="s1">&#39;green&#39;</span><span class="p">,</span>
                                       <span class="n">permutation_spatial</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                                       <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">ax</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;HMDB0000073&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>

<span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_group_communication_heatmap</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                                       <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                                       <span class="n">group_name</span> <span class="o">=</span> <span class="s2">&quot;tissue_type&quot;</span><span class="p">,</span>
                                       <span class="n">metabolite_name</span> <span class="o">=</span> <span class="s1">&#39;HMDB0000216&#39;</span><span class="p">,</span>
                                       <span class="n">cmap</span> <span class="o">=</span> <span class="s1">&#39;blue&#39;</span><span class="p">,</span>
                                       <span class="n">permutation_spatial</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                                       <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
<span class="n">ax</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;HMDB0000216&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Text(0.5, 1.0, &#39;HMDB0000216&#39;)
</pre></div>
</div>
<img alt="../_images/2cbaf359dd5d20aac3d9b0ba5f98f2295079314162c81b5e4ec146857b624831.png" src="../_images/2cbaf359dd5d20aac3d9b0ba5f98f2295079314162c81b5e4ec146857b624831.png" />
</div>
</div>
</section>
<section id="mcc-remodelling-in-receiver-cells">
<h3>MCC remodelling in receiver cells<a class="headerlink" href="#mcc-remodelling-in-receiver-cells" title="Link to this heading"></a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">adata_raw</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="s1">&#39;data/multiomics_mousebrain_parkinson_RNA.h5ad&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">metabolite_name</span> <span class="o">=</span> <span class="s1">&#39;HMDB0000073&#39;</span>
<span class="n">df_deg</span><span class="p">,</span> <span class="n">df_yhat</span> <span class="o">=</span> <span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">communication_responseGenes</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">,</span>
                                                    <span class="n">adata_raw</span> <span class="o">=</span> <span class="n">adata_raw</span><span class="p">,</span>
                                                    <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                                                    <span class="n">metabolite_name</span> <span class="o">=</span> <span class="n">metabolite_name</span><span class="p">,</span>
                                                    <span class="n">summary</span> <span class="o">=</span> <span class="s1">&#39;receiver&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>  |                                                  | 0 % ~calculating   |+                                                 | 1 % ~58m 04s       |++                                                | 2 % ~30m 13s       |++                                                | 3 % ~21m 09s       |+++                                               | 4 % ~16m 21s       |+++                                               | 5 % ~13m 31s       |++++                                              | 6 % ~11m 37s       |++++                                              | 7 % ~10m 20s       |+++++                                             | 8 % ~09m 23s       |+++++                                             | 9 % ~08m 31s       |++++++                                            | 10% ~07m 51s       |++++++                                            | 11% ~07m 17s       |+++++++                                           | 12% ~06m 49s       |+++++++                                           | 13% ~06m 25s       |++++++++                                          | 14% ~06m 05s       |++++++++                                          | 15% ~05m 50s       |+++++++++                                         | 16% ~05m 34s       |+++++++++                                         | 17% ~05m 20s       |++++++++++                                        | 18% ~05m 07s       |++++++++++                                        | 19% ~04m 54s       |+++++++++++                                       | 20% ~04m 43s       |+++++++++++                                       | 21% ~04m 32s       |++++++++++++                                      | 22% ~04m 22s       |++++++++++++                                      | 23% ~04m 16s       |+++++++++++++                                     | 24% ~04m 08s       |+++++++++++++                                     | 25% ~04m 01s       |++++++++++++++                                    | 26% ~03m 53s       |++++++++++++++                                    | 27% ~03m 47s       |+++++++++++++++                                   | 28% ~03m 40s       |+++++++++++++++                                   | 29% ~03m 34s       |++++++++++++++++                                  | 30% ~03m 31s       |++++++++++++++++                                  | 31% ~03m 25s       |+++++++++++++++++                                 | 32% ~03m 19s       |+++++++++++++++++                                 | 33% ~03m 13s       |++++++++++++++++++                                | 34% ~03m 09s       |++++++++++++++++++                                | 35% ~03m 04s       |+++++++++++++++++++                               | 36% ~02m 59s       |+++++++++++++++++++                               | 37% ~02m 54s       |++++++++++++++++++++                              | 38% ~02m 50s       |++++++++++++++++++++                              | 39% ~02m 46s       |+++++++++++++++++++++                             | 40% ~02m 41s       |+++++++++++++++++++++                             | 41% ~02m 39s       |++++++++++++++++++++++                            | 42% ~02m 35s       |++++++++++++++++++++++                            | 43% ~02m 31s       |+++++++++++++++++++++++                           | 44% ~02m 27s       |+++++++++++++++++++++++                           | 45% ~02m 23s       |++++++++++++++++++++++++                          | 46% ~02m 20s       |++++++++++++++++++++++++                          | 47% ~02m 16s       |+++++++++++++++++++++++++                         | 48% ~02m 13s       |+++++++++++++++++++++++++                         | 49% ~02m 09s       |++++++++++++++++++++++++++                        | 51% ~02m 06s       |++++++++++++++++++++++++++                        | 52% ~02m 03s       |+++++++++++++++++++++++++++                       | 53% ~01m 60s       |+++++++++++++++++++++++++++                       | 54% ~01m 57s       |++++++++++++++++++++++++++++                      | 55% ~01m 54s       |++++++++++++++++++++++++++++                      | 56% ~01m 51s       |+++++++++++++++++++++++++++++                     | 57% ~01m 48s       |+++++++++++++++++++++++++++++                     | 58% ~01m 45s       |++++++++++++++++++++++++++++++                    | 59% ~01m 42s       |++++++++++++++++++++++++++++++                    | 60% ~01m 39s       |+++++++++++++++++++++++++++++++                   | 61% ~01m 36s       |+++++++++++++++++++++++++++++++                   | 62% ~01m 33s       |++++++++++++++++++++++++++++++++                  | 63% ~01m 31s       |++++++++++++++++++++++++++++++++                  | 64% ~01m 28s       |+++++++++++++++++++++++++++++++++                 | 65% ~01m 25s       |+++++++++++++++++++++++++++++++++                 | 66% ~01m 22s       |++++++++++++++++++++++++++++++++++                | 67% ~01m 19s       |++++++++++++++++++++++++++++++++++                | 68% ~01m 17s       |+++++++++++++++++++++++++++++++++++               | 69% ~01m 14s       |+++++++++++++++++++++++++++++++++++               | 70% ~01m 11s       |++++++++++++++++++++++++++++++++++++              | 71% ~01m 09s       |++++++++++++++++++++++++++++++++++++              | 72% ~01m 07s       |+++++++++++++++++++++++++++++++++++++             | 73% ~01m 04s       |+++++++++++++++++++++++++++++++++++++             | 74% ~01m 02s       |++++++++++++++++++++++++++++++++++++++            | 75% ~60s           |++++++++++++++++++++++++++++++++++++++            | 76% ~57s           |+++++++++++++++++++++++++++++++++++++++           | 77% ~54s           |+++++++++++++++++++++++++++++++++++++++           | 78% ~52s           |++++++++++++++++++++++++++++++++++++++++          | 79% ~49s           |++++++++++++++++++++++++++++++++++++++++          | 80% ~47s           |+++++++++++++++++++++++++++++++++++++++++         | 81% ~44s           |+++++++++++++++++++++++++++++++++++++++++         | 82% ~42s           |++++++++++++++++++++++++++++++++++++++++++        | 83% ~39s           |++++++++++++++++++++++++++++++++++++++++++        | 84% ~37s           |+++++++++++++++++++++++++++++++++++++++++++       | 85% ~35s           |+++++++++++++++++++++++++++++++++++++++++++       | 86% ~32s           |++++++++++++++++++++++++++++++++++++++++++++      | 87% ~30s           |++++++++++++++++++++++++++++++++++++++++++++      | 88% ~28s           |+++++++++++++++++++++++++++++++++++++++++++++     | 89% ~25s           |+++++++++++++++++++++++++++++++++++++++++++++     | 90% ~23s           |++++++++++++++++++++++++++++++++++++++++++++++    | 91% ~21s           |++++++++++++++++++++++++++++++++++++++++++++++    | 92% ~18s           |+++++++++++++++++++++++++++++++++++++++++++++++   | 93% ~16s           |+++++++++++++++++++++++++++++++++++++++++++++++   | 94% ~14s           |++++++++++++++++++++++++++++++++++++++++++++++++  | 95% ~11s           |++++++++++++++++++++++++++++++++++++++++++++++++  | 96% ~09s           |+++++++++++++++++++++++++++++++++++++++++++++++++ | 97% ~07s           |+++++++++++++++++++++++++++++++++++++++++++++++++ | 98% ~05s           |++++++++++++++++++++++++++++++++++++++++++++++++++| 99% ~02s           |++++++++++++++++++++++++++++++++++++++++++++++++++| 100% elapsed=03m 42s
2 parameter combinations, 2 use sequential method, 2 use subsampling method
Running Clustering on Parameter Combinations...
done.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pickle</span>
<span class="n">deg_result</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;df_deg&quot;</span><span class="p">:</span> <span class="n">df_deg</span><span class="p">,</span> <span class="s2">&quot;df_yhat&quot;</span><span class="p">:</span> <span class="n">df_yhat</span><span class="p">}</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;results/responseGenes_&quot;</span> <span class="o">+</span> <span class="n">metabolite_name</span> <span class="o">+</span> <span class="s2">&quot;.pkl&quot;</span><span class="p">,</span> <span class="s1">&#39;wb&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">handle</span><span class="p">:</span>
    <span class="n">pickle</span><span class="o">.</span><span class="n">dump</span><span class="p">(</span><span class="n">deg_result</span><span class="p">,</span> <span class="n">handle</span><span class="p">,</span> <span class="n">protocol</span><span class="o">=</span><span class="n">pickle</span><span class="o">.</span><span class="n">HIGHEST_PROTOCOL</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">metabolite_name</span> <span class="o">=</span> <span class="s1">&#39;HMDB0000073&#39;</span>
<span class="kn">import</span> <span class="nn">pickle</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;results/responseGenes_&quot;</span> <span class="o">+</span> <span class="n">metabolite_name</span> <span class="o">+</span> <span class="s2">&quot;.pkl&quot;</span><span class="p">,</span> <span class="s1">&#39;rb&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">file</span><span class="p">:</span>
    <span class="n">deg_result</span> <span class="o">=</span> <span class="n">pickle</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">file</span><span class="p">)</span>
<span class="n">df_deg</span> <span class="o">=</span> <span class="n">deg_result</span><span class="p">[</span><span class="s2">&quot;df_deg&quot;</span><span class="p">]</span>
<span class="n">df_yhat</span> <span class="o">=</span> <span class="n">deg_result</span><span class="p">[</span><span class="s2">&quot;df_yhat&quot;</span><span class="p">]</span>
<span class="n">df_deg_clus</span><span class="p">,</span> <span class="n">df_yhat_clus</span> <span class="o">=</span> <span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">communication_responseGenes_cluster</span><span class="p">(</span><span class="n">df_deg</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,:</span><span class="mi">3</span><span class="p">],</span> <span class="n">df_yhat</span><span class="p">,</span> <span class="n">deg_clustering_res</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_communication_responseGenes</span><span class="p">(</span><span class="n">df_deg</span> <span class="o">=</span> <span class="n">df_deg_clus</span><span class="p">,</span>
                                       <span class="n">df_yhat</span> <span class="o">=</span> <span class="n">df_yhat_clus</span><span class="p">,</span>
                                       <span class="n">top_ngene_per_cluster</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
                                       <span class="n">colormap</span> <span class="o">=</span> <span class="s1">&#39;viridis&#39;</span><span class="p">,</span>
                                       <span class="n">color_range</span> <span class="o">=</span> <span class="p">(</span><span class="o">-</span><span class="mi">2</span><span class="p">,</span><span class="mi">2</span><span class="p">),</span>
                                       <span class="n">font_scale</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span>
                                       <span class="n">figsize</span> <span class="o">=</span> <span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">6</span><span class="p">),</span>
                                       <span class="n">return_genes</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/fa1ad1a80da265b0f79a8fead9879369696c700f868d82384d6f50f04d7f32ac.png" src="../_images/fa1ad1a80da265b0f79a8fead9879369696c700f868d82384d6f50f04d7f32ac.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">df_result</span> <span class="o">=</span> <span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">communication_responseGenes_keggEnrich</span><span class="p">(</span><span class="n">gene_list</span> <span class="o">=</span> <span class="n">df_deg</span><span class="p">[</span><span class="n">df_deg</span><span class="p">[</span><span class="s1">&#39;pvalue&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mf">0.05</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">tolist</span><span class="p">()[:</span><span class="mi">1000</span><span class="p">],</span>
                                                         <span class="n">gene_sets</span> <span class="o">=</span> <span class="s2">&quot;KEGG_2019_Mouse&quot;</span><span class="p">,</span>
                                                         <span class="n">organism</span> <span class="o">=</span> <span class="s2">&quot;Mouse&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">df_result</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;results/responseGenes_keggEnrich_&quot;</span> <span class="o">+</span> <span class="n">metabolite_name</span> <span class="o">+</span> <span class="s2">&quot;.csv&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_communication_responseGenes_keggEnrich</span><span class="p">(</span><span class="n">df_result</span> <span class="o">=</span> <span class="n">df_result</span><span class="p">,</span>
                                                  <span class="n">show_term_order</span> <span class="o">=</span> <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">7</span><span class="p">,</span><span class="mi">9</span><span class="p">],</span>
                                                  <span class="n">cmap</span> <span class="o">=</span> <span class="s1">&#39;blue&#39;</span><span class="p">,</span>
                                                  <span class="n">maxshow_gene</span> <span class="o">=</span> <span class="mi">10</span><span class="p">,</span>
                                                  <span class="n">figsize</span> <span class="o">=</span> <span class="p">(</span><span class="mi">7</span><span class="p">,</span><span class="mi">5</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/a429d00005335bc8bdfd52911297182df3357265cf66f01e98631940a96e1e52.png" src="../_images/a429d00005335bc8bdfd52911297182df3357265cf66f01e98631940a96e1e52.png" />
</div>
</div>
</section>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="toy_example1.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">Toy example 1</div>
              </div>
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="../tutorials.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">&lt;no title&gt;</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2024, Songhao Luo
            </div>
            Made with 
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            <div class="icons">
              <a class="muted-link fab fa-github" href="https://github.com/SonghaoLuo/metachat" aria-label="GitHub"></a>
              
            </div>
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Example of spatial multi-omics data (mouse brain with Parkinson’s disease)</a><ul>
<li><a class="reference internal" href="#data-preprocessing">Data preprocessing</a><ul>
<li><a class="reference internal" href="#preprocessing-for-transcriptomics-data">Preprocessing for transcriptomics data</a></li>
<li><a class="reference internal" href="#preprocessing-for-metabolomics-data">Preprocessing for metabolomics data</a></li>
<li><a class="reference internal" href="#combine-the-transcriptomics-and-metabolomics-data">Combine the transcriptomics and metabolomics data</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mcc-inference">MCC Inference</a><ul>
<li><a class="reference internal" href="#import-metachatdb">Import MetaChatDB</a></li>
<li><a class="reference internal" href="#identify-long-range-channels-lrc">Identify long-range channels (LRC)</a></li>
</ul>
</li>
<li><a class="reference internal" href="#downstream-analysis">Downstream analysis</a><ul>
<li><a class="reference internal" href="#summary-communication-among-different-scales">Summary communication among different scales</a></li>
<li><a class="reference internal" href="#identifing-mcc-flow">Identifing MCC flow</a></li>
<li><a class="reference internal" href="#constructing-group-level-mcc">Constructing group-level MCC</a></li>
<li><a class="reference internal" href="#mcc-remodelling-in-receiver-cells">MCC remodelling in receiver cells</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="../_static/jquery.js?v=5d32c60e"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
    <script src="../_static/documentation_options.js?v=e3a6060d"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/scripts/furo.js?v=4e2eecee"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    </body>
</html>