<!doctype html>
<html class="no-js" lang="en" data-content_root="../">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="../genindex.html" /><link rel="search" title="Search" href="../search.html" /><link rel="next" title="Toy example 1" href="toy_example1.html" /><link rel="prev" title="&lt;no title&gt;" href="../tutorials.html" />

    <!-- Generated with Sphinx 7.3.7 and Furo 2024.05.06 -->
        <title>Example of spatial multi-omics data (mouse brain with Parkinson’s disease) - MetaChat 0.0.2 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=a746c00c" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo.css?v=387cc868" />
    <link rel="stylesheet" type="text/css" href="../_static/css/sphinx_rtd_size.css?v=8f58cd29" />
    <link rel="stylesheet" type="text/css" href="../_static/sg_gallery.css?v=61a4c737" />
    <link rel="stylesheet" type="text/css" href="../_static/nbsphinx-code-cells.css?v=2aa19091" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo-extensions.css?v=36a5483c" />
    <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css" />
    <link rel="stylesheet" type="text/css" href="../_static/css/override.css?v=46f5334b" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --color-brand-primary: #357473;
  --color-brand-content: #357473;
  --admonition-font-size: var(--font-size-normal);
  --admonition-title-font-size: var(--font-size-normal);
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
  <symbol id="svg-sun-with-moon" viewBox="0 0 24 24">
    <title>Auto light/dark, in light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path style="opacity: 50%" d="M 5.411 14.504 C 5.471 14.504 5.532 14.504 5.591 14.504 C 3.639 16.319 4.383 19.569 6.931 20.352 C 7.693 20.586 8.512 20.551 9.25 20.252 C 8.023 23.207 4.056 23.725 2.11 21.184 C 0.166 18.642 1.702 14.949 4.874 14.536 C 5.051 14.512 5.231 14.5 5.411 14.5 L 5.411 14.504 Z"/>
      <line x1="14.5" y1="3.25" x2="14.5" y2="1.25"/>
      <line x1="14.5" y1="15.85" x2="14.5" y2="17.85"/>
      <line x1="10.044" y1="5.094" x2="8.63" y2="3.68"/>
      <line x1="19" y1="14.05" x2="20.414" y2="15.464"/>
      <line x1="8.2" y1="9.55" x2="6.2" y2="9.55"/>
      <line x1="20.8" y1="9.55" x2="22.8" y2="9.55"/>
      <line x1="10.044" y1="14.006" x2="8.63" y2="15.42"/>
      <line x1="19" y1="5.05" x2="20.414" y2="3.636"/>
      <circle cx="14.5" cy="9.55" r="3.6"/>
    </svg>
  </symbol>
  <symbol id="svg-moon-with-sun" viewBox="0 0 24 24">
    <title>Auto light/dark, in dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path d="M 8.282 7.007 C 8.385 7.007 8.494 7.007 8.595 7.007 C 5.18 10.184 6.481 15.869 10.942 17.24 C 12.275 17.648 13.706 17.589 15 17.066 C 12.851 22.236 5.91 23.143 2.505 18.696 C -0.897 14.249 1.791 7.786 7.342 7.063 C 7.652 7.021 7.965 7 8.282 7 L 8.282 7.007 Z"/>
      <line style="opacity: 50%" x1="18" y1="3.705" x2="18" y2="2.5"/>
      <line style="opacity: 50%" x1="18" y1="11.295" x2="18" y2="12.5"/>
      <line style="opacity: 50%" x1="15.316" y1="4.816" x2="14.464" y2="3.964"/>
      <line style="opacity: 50%" x1="20.711" y1="10.212" x2="21.563" y2="11.063"/>
      <line style="opacity: 50%" x1="14.205" y1="7.5" x2="13.001" y2="7.5"/>
      <line style="opacity: 50%" x1="21.795" y1="7.5" x2="23" y2="7.5"/>
      <line style="opacity: 50%" x1="15.316" y1="10.184" x2="14.464" y2="11.036"/>
      <line style="opacity: 50%" x1="20.711" y1="4.789" x2="21.563" y2="3.937"/>
      <circle style="opacity: 50%" cx="18" cy="7.5" r="2.169"/>
    </svg>
  </symbol>
  <symbol id="svg-pencil" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-pencil-code">
      <path d="M4 20h4l10.5 -10.5a2.828 2.828 0 1 0 -4 -4l-10.5 10.5v4" />
      <path d="M13.5 6.5l4 4" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" /
    </svg>
  </symbol>
  <symbol id="svg-eye" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-eye-code">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0" />
      <path
        d="M11.11 17.958c-3.209 -.307 -5.91 -2.293 -8.11 -5.958c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6c-.21 .352 -.427 .688 -.647 1.008" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>

<a class="skip-to-content muted-link" href="#furo-main-content">Skip to content</a>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../index.html"><div class="brand">MetaChat 0.0.2 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
          <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo" src="../_static/logo.png" alt="Logo"/>
  </div>
  
  
</a><form class="sidebar-search-container" method="get" action="../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">Example of spatial multi-omics data (mouse brain with Parkinson’s disease)</a></li>
<li class="toctree-l1"><a class="reference internal" href="toy_example1.html">Toy example 1</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../api/index.html">API</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle navigation of API</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pp.MetaChatDB.html">metachat.pp.MetaChatDB</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pp.LRC_unfiltered.html">metachat.pp.LRC_unfiltered</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pp.LRC_cluster.html">metachat.pp.LRC_cluster</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pp.LRC_filtered.html">metachat.pp.LRC_filtered</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pp.compute_costDistance.html">metachat.pp.compute_costDistance</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.metabolic_communication.html">metachat.tl.metabolic_communication</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.summary_communication.html">metachat.tl.summary_communication</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_flow.html">metachat.tl.communication_flow</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_group.html">metachat.tl.communication_group</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_group_spatial.html">metachat.tl.communication_group_spatial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.summary_pathway.html">metachat.tl.summary_pathway</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_responseGenes.html">metachat.tl.communication_responseGenes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_responseGenes_cluster.html">metachat.tl.communication_responseGenes_cluster</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.tl.communication_responseGenes_keggEnrich.html">metachat.tl.communication_responseGenes_keggEnrich</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_communication_flow.html">metachat.pl.plot_communication_flow</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_group_communication_chord.html">metachat.pl.plot_group_communication_chord</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_group_communication_heatmap.html">metachat.pl.plot_group_communication_heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_communication_responseGenes.html">metachat.pl.plot_communication_responseGenes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_group_communication_compare_hierarchy_diagram.html">metachat.pl.plot_group_communication_compare_hierarchy_diagram</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_MSpair_contribute_group.html">metachat.pl.plot_MSpair_contribute_group</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_communication_responseGenes_keggEnrich.html">metachat.pl.plot_communication_responseGenes_keggEnrich</a></li>
<li class="toctree-l2"><a class="reference internal" href="../api/metachat.pl.plot_summary_pathway.html">metachat.pl.plot_summary_pathway</a></li>
</ul>
</li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          

<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
              <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main" id="furo-main-content">
          <section id="Example-of-spatial-multi-omics-data-(mouse-brain-with-Parkinson's-disease)">
<h1>Example of spatial multi-omics data (mouse brain with Parkinson’s disease)<a class="headerlink" href="#Example-of-spatial-multi-omics-data-(mouse-brain-with-Parkinson's-disease)" title="Link to this heading"></a></h1>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>import os
import numpy as np
import pandas as pd
import scanpy as sc
import squidpy as sq
import seaborn as sns
import matplotlib.pyplot as plt

import metachat as mc
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span># setting work dictionary
os.chdir(&quot;/home/Project/MetaChat_test/mouse_brain&quot;)
</pre></div>
</div>
</div>
<section id="Data-preprocessing">
<h2>Data preprocessing<a class="headerlink" href="#Data-preprocessing" title="Link to this heading"></a></h2>
<section id="Preprocessing-for-transcriptomics-data">
<h3>Preprocessing for transcriptomics data<a class="headerlink" href="#Preprocessing-for-transcriptomics-data" title="Link to this heading"></a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>adata_RNA = sc.read_h5ad(&#39;data/multiomics_mousebrain_parkinson_RNA.h5ad&#39;)
adata_RNA
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
AnnData object with n_obs × n_vars = 3011 × 32285
    obs: &#39;in_tissue&#39;, &#39;array_row&#39;, &#39;array_col&#39;
    var: &#39;gene_ids&#39;, &#39;feature_types&#39;, &#39;genome&#39;
    uns: &#39;spatial&#39;
    obsm: &#39;spatial&#39;
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>adata_RNA.var_names_make_unique()
adata_RNA.raw = adata_RNA
sc.pp.normalize_total(adata_RNA, inplace=True)
sc.pp.log1p(adata_RNA)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>adata_visual = adata_RNA.copy()
random_seed = 42
sc.pp.highly_variable_genes(adata_visual, min_mean=0.0125, max_mean=10, min_disp=0.2)
adata_visual = adata_visual[:, adata_visual.var.highly_variable]
sc.tl.pca(adata_visual, svd_solver=&#39;arpack&#39;, random_state=random_seed)
sc.pp.neighbors(adata_visual, n_neighbors=10, n_pcs=40, random_state=random_seed)
sc.tl.umap(adata_visual, random_state=random_seed)
sc.tl.leiden(adata_visual, resolution=1)
sc.pl.spatial(adata_visual, color=&#39;leiden&#39;)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_7_0.png" src="../_images/notebooks_Multi_omics_data_7_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>cluster_to_cell_type = {
    &#39;0&#39;: &#39;Striatum dorsal region&#39;,
    &#39;1&#39;: &#39;Cerebral cortex&#39;,
    &#39;2&#39;: &#39;Striatum ventral region&#39;,
    &#39;3&#39;: &#39;Cerebral cortex&#39;,
    &#39;4&#39;: &#39;Lateral septal complex&#39;,
    &#39;5&#39;: &#39;Fiber tracts&#39;,
    &#39;6&#39;: &#39;Cerebral cortex&#39;,
    &#39;7&#39;: &#39;Fiber tracts&#39;,
    &#39;8&#39;: &#39;Cerebral cortex&#39;,
    &#39;9&#39;: &#39;Cerebral cortex&#39;,
    &#39;10&#39;: &#39;Striatum dorsal region&#39;,
    &#39;11&#39;: &#39;Striatum ventral region&#39;,
    &#39;12&#39;: &#39;Ventricular systems&#39;,
    &#39;13&#39;: &#39;Cerebral cortex&#39;,
}
adata_RNA.obs[&#39;tissue_type&#39;] = adata_visual.obs[&#39;leiden&#39;].map(cluster_to_cell_type)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>fig, ax = plt.subplots(figsize = (5,5))
sq.pl.spatial_scatter(adata = adata_RNA,
                      color = &quot;tissue_type&quot;,
                      ax = ax)
ax.set_box_aspect(1)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_9_0.png" src="../_images/notebooks_Multi_omics_data_9_0.png" />
</div>
</div>
</section>
<section id="Preprocessing-for-metabolomics-data">
<h3>Preprocessing for metabolomics data<a class="headerlink" href="#Preprocessing-for-metabolomics-data" title="Link to this heading"></a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>adata_MSI = sc.read_h5ad(&#39;data/multiomics_mousebrain_parkinson_MSI.h5ad&#39;)
adata_MSI.uns[&#39;spatial&#39;] = adata_RNA.uns[&#39;spatial&#39;]
adata_MSI.obsm[&#39;spatial&#39;] = adata_RNA.obsm[&#39;spatial&#39;]
adata_MSI
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
AnnData object with n_obs × n_vars = 3011 × 3015
    obs: &#39;orig.ident&#39;, &#39;nCount_RNA&#39;, &#39;nFeature_RNA&#39;
    var: &#39;features&#39;
    uns: &#39;spatial&#39;
    obsm: &#39;spatial&#39;
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>adata_MSI.var_names_make_unique()
adata_RNA.raw = adata_RNA
sc.pp.normalize_total(adata_MSI, inplace=True)
</pre></div>
</div>
</div>
</section>
<section id="Combine-the-transcriptomics-and-metabolomics-data">
<h3>Combine the transcriptomics and metabolomics data<a class="headerlink" href="#Combine-the-transcriptomics-and-metabolomics-data" title="Link to this heading"></a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>from scipy.sparse import hstack
X_combined = hstack([adata_RNA.X, adata_MSI.X])
X_combined = X_combined.tocsr()
var_combined = pd.DataFrame({&quot;features&quot; : np.concatenate((np.array(adata_RNA.var_names),np.array(adata_MSI.var_names)))})
var_combined.index = var_combined[&quot;features&quot;]
adata = sc.AnnData(X_combined, obs=adata_RNA.obs, var=var_combined)
adata.obsm = adata_RNA.obsm
adata.uns = adata_RNA.uns
</pre></div>
</div>
</div>
</section>
</section>
<section id="MCC-Inference">
<h2>MCC Inference<a class="headerlink" href="#MCC-Inference" title="Link to this heading"></a></h2>
<section id="Import-MetaChatDB">
<h3>Import MetaChatDB<a class="headerlink" href="#Import-MetaChatDB" title="Link to this heading"></a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>df_metasen = mc.pp.MetaChatDB(species=&#39;mouse&#39;)
</pre></div>
</div>
</div>
</section>
<section id="Identify-long-range-channels">
<h3>Identify long-range channels<a class="headerlink" href="#Identify-long-range-channels" title="Link to this heading"></a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>LRC_name = &quot;CSF&quot;
LRC_marker_gene = &quot;Gfap&quot;

fig, ax = plt.subplots(figsize = (6,5))
sq.pl.spatial_scatter(adata,
                      color = [LRC_marker_gene],
                      ax = ax)
ax.set_box_aspect(1)
# plt.savefig(&quot;figures/&quot; + &quot;LRC_&quot; + LRC_name + &quot;_&quot; + LRC_marker_gene + &quot;.pdf&quot;)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_19_0.png" src="../_images/notebooks_Multi_omics_data_19_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>mc.pp.LRC_unfiltered(adata = adata,
                     LRC_name = LRC_name,
                     LRC_marker_gene = LRC_marker_gene,
                     quantile = 85)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Cells with Gfap expression above the 85% have been selected as candidates and stored in &#39;adata.obs[&#39;LRC_CSF_unfiltered&#39;]&#39;.
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>fig, ax = plt.subplots(figsize = (6,5))
sq.pl.spatial_scatter(adata,
                      color = &#39;LRC_&#39; + LRC_name + &#39;_unfiltered&#39;,
                      ax = ax)
ax.set_box_aspect(1)
# plt.savefig(&quot;figures/&quot; + &quot;LRC_&quot; + LRC_name + &quot;_unfiltered.pdf&quot;)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_21_0.png" src="../_images/notebooks_Multi_omics_data_21_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span># There is a clear long-range channel as CSF vessel, so we perform density peak clustering on points with high marker expression in CSF vessel.
LRC_cluster_CSF = mc.pp.LRC_cluster(adata = adata,
                                    LRC_name = LRC_name,
                                    density_cutoff = 7,
                                    delta_cutoff = 8000,
                                    outlier_cutoff = 4.5)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_22_0.png" src="../_images/notebooks_Multi_omics_data_22_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>mc.pp.LRC_filtered(adata = adata,
                   LRC_name = LRC_name,
                   LRC_cluster = LRC_cluster_CSF)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Candidate points for CSF LRC are clustered and outliers are removed. LRC points are stored in &#39;adata.obs[&#39;LRC_CSF_filtered&#39;]&#39;.
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>import matplotlib.colors as mcolors
fig, ax = plt.subplots(figsize = (6,5))
sq.pl.spatial_scatter(adata,
                      color = &#39;LRC_&#39; + LRC_name + &#39;_filtered&#39;,
                      alpha = 1,
                      ax = ax)
ax.set_box_aspect(1)
# plt.savefig(&quot;figures/&quot; + &quot;LRC_&quot; + LRC_name + &quot;_filtered.pdf&quot;)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_24_0.png" src="../_images/notebooks_Multi_omics_data_24_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>mc.pp.compute_costDistance(adata = adata,
                           LRC_type = [&quot;CSF&quot;],
                           dis_thr = 1000,
                           k_neighb = 5,
                           LRC_strength = 2000,
                           plot = True)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Compute spatial cost distance without long-range channel...
Compute spatial cost distance incorporating long-range channel of CSF
The points that can occur long-range communication is...
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_25_1.png" src="../_images/notebooks_Multi_omics_data_25_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
The points that can occur long-range communication is...
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_25_3.png" src="../_images/notebooks_Multi_omics_data_25_3.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
  Construct network graph of long-range channel among 5 neighborhoods...
  Calculate the shortest path distance from the source to the target using the shortest path algorithm...
    For the long-range case of cluster 1...
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 190/190 [00:14&lt;00:00, 13.05it/s]
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
    For the long-range case of cluster 2...
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 14/14 [00:00&lt;00:00, 1925.51it/s]
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
  Rearrange distance matrix...
    For the long-range case of cluster 1...
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 448/448 [00:00&lt;00:00, 676.06it/s]
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
    For the long-range case of cluster 2...
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 36/36 [00:00&lt;00:00, 8275.06it/s]
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Finished!
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>mc.tl.metabolic_communication(adata = adata,
                              database_name = &#39;MetaChatDB&#39;,
                              df_metasen = df_metasen,
                              LRC_type = [&quot;CSF&quot;],
                              dis_thr = 1000)
adata.write(&#39;results/inference_results.h5ad&#39;)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
There are 327 pairs were found from the spatial data.
</pre></div></div>
</div>
</section>
</section>
<section id="Downstream-analysis">
<h2>Downstream analysis<a class="headerlink" href="#Downstream-analysis" title="Link to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>adata = sc.read(&quot;results/inference_results.h5ad&quot;)
</pre></div>
</div>
</div>
<section id="Summary-communication-among-different-scales">
<h3>Summary communication among different scales<a class="headerlink" href="#Summary-communication-among-different-scales" title="Link to this heading"></a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>sum_metabolites = [&#39;HMDB0000216&#39;,&#39;HMDB0000073&#39;]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>mc.tl.summary_communication(adata = adata,
                            database_name = &#39;MetaChatDB&#39;,
                            sum_metabolites = sum_metabolites)
</pre></div>
</div>
</div>
</section>
<section id="Identifing-MCC-flow">
<h3>Identifing MCC flow<a class="headerlink" href="#Identifing-MCC-flow" title="Link to this heading"></a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>mc.tl.communication_flow(adata = adata,
                         database_name = &#39;MetaChatDB&#39;,
                         sum_metabolites = sum_metabolites,
                         k = 5)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>fig, ax = plt.subplots(1, 3, figsize=(15,5))
mc.pl.plot_communication_flow(adata = adata,
                              database_name = &#39;MetaChatDB&#39;,
                              metabolite_name = &#39;HMDB0000073&#39;,
                              plot_method = &#39;grid&#39;,
                              background = &#39;image&#39;,
                              group_name = &#39;tissue_type&#39;,
                              summary = &#39;sender&#39;,
                              ndsize = 8,
                              scale = 0.0000000000003,
                              normalize_v_quantile = 0.995,
                              grid_density = 0.4,
                              title = &#39;Image&#39;,
                              ax = ax[0])
ax[0].set_box_aspect(1)

mc.pl.plot_communication_flow(adata = adata,
                              database_name = &#39;MetaChatDB&#39;,
                              metabolite_name = &#39;HMDB0000073&#39;,
                              plot_method = &#39;grid&#39;,
                              background = &#39;group&#39;,
                              group_name = &#39;tissue_type&#39;,
                              summary = &#39;sender&#39;,
                              ndsize = 8,
                              scale = 0.0000000000003,
                              normalize_v_quantile = 0.995,
                              grid_density = 0.4,
                              title = &#39;Group types&#39;,
                              ax = ax[1])
ax[1].set_box_aspect(1)

mc.pl.plot_communication_flow(adata = adata,
                              database_name = &#39;MetaChatDB&#39;,
                              metabolite_name = &#39;HMDB0000073&#39;,
                              plot_method = &#39;grid&#39;,
                              background = &#39;summary&#39;,
                              group_name = &#39;tissue_type&#39;,
                              summary = &#39;sender&#39;,
                              cmap = &#39;YlOrRd&#39;,
                              ndsize = 8,
                              scale = 0.0000000000003,
                              normalize_v_quantile = 0.995,
                              grid_density = 0.4,
                              title = &#39;MCC Signals&#39;,
                              ax = ax[2])
ax[2].set_box_aspect(1)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_34_0.png" src="../_images/notebooks_Multi_omics_data_34_0.png" />
</div>
</div>
</section>
<section id="Constructing-group-level-MCC">
<h3>Constructing group-level MCC<a class="headerlink" href="#Constructing-group-level-MCC" title="Link to this heading"></a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>mc.tl.communication_group(adata = adata,
                          database_name = &#39;MetaChatDB&#39;,
                          group_name = &quot;tissue_type&quot;,
                          sum_metabolites = sum_metabolites,
                          n_permutations = 100)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>adata.uns[&#39;MetaChat_group-tissue_type-MetaChatDB-HMDB0000073&#39;][&#39;communication_matrix&#39;]
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Cerebral cortex</th>
      <th>Fiber tracts</th>
      <th>Lateral septal complex</th>
      <th>Striatum dorsal region</th>
      <th>Striatum ventral region</th>
      <th>Ventricular systems</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Cerebral cortex</th>
      <td>1.060052e-11</td>
      <td>5.985997e-12</td>
      <td>5.367747e-12</td>
      <td>3.851556e-12</td>
      <td>5.980569e-12</td>
      <td>1.108535e-11</td>
    </tr>
    <tr>
      <th>Fiber tracts</th>
      <td>8.941187e-12</td>
      <td>1.756259e-11</td>
      <td>1.813045e-17</td>
      <td>1.101182e-11</td>
      <td>1.427381e-11</td>
      <td>2.586341e-11</td>
    </tr>
    <tr>
      <th>Lateral septal complex</th>
      <td>1.827025e-12</td>
      <td>5.856315e-13</td>
      <td>1.715571e-11</td>
      <td>3.597803e-13</td>
      <td>3.308491e-13</td>
      <td>1.257162e-12</td>
    </tr>
    <tr>
      <th>Striatum dorsal region</th>
      <td>5.726325e-12</td>
      <td>1.436869e-11</td>
      <td>8.287509e-18</td>
      <td>3.606240e-10</td>
      <td>3.068785e-11</td>
      <td>1.409258e-11</td>
    </tr>
    <tr>
      <th>Striatum ventral region</th>
      <td>2.451099e-12</td>
      <td>3.763631e-12</td>
      <td>3.598783e-18</td>
      <td>1.658457e-11</td>
      <td>2.576112e-10</td>
      <td>3.478752e-12</td>
    </tr>
    <tr>
      <th>Ventricular systems</th>
      <td>7.851767e-12</td>
      <td>2.412584e-11</td>
      <td>1.282194e-17</td>
      <td>9.945576e-12</td>
      <td>1.285908e-11</td>
      <td>4.135405e-10</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>adata.uns[&#39;MetaChat_group-tissue_type-MetaChatDB-HMDB0000073&#39;][&#39;communication_pvalue&#39;]
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Cerebral cortex</th>
      <th>Fiber tracts</th>
      <th>Lateral septal complex</th>
      <th>Striatum dorsal region</th>
      <th>Striatum ventral region</th>
      <th>Ventricular systems</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Cerebral cortex</th>
      <td>1.00</td>
      <td>1.00</td>
      <td>0.99</td>
      <td>1.00</td>
      <td>0.99</td>
      <td>0.37</td>
    </tr>
    <tr>
      <th>Fiber tracts</th>
      <td>0.93</td>
      <td>1.00</td>
      <td>1.00</td>
      <td>0.48</td>
      <td>0.18</td>
      <td>0.13</td>
    </tr>
    <tr>
      <th>Lateral septal complex</th>
      <td>1.00</td>
      <td>1.00</td>
      <td>1.00</td>
      <td>1.00</td>
      <td>1.00</td>
      <td>0.78</td>
    </tr>
    <tr>
      <th>Striatum dorsal region</th>
      <td>1.00</td>
      <td>0.18</td>
      <td>1.00</td>
      <td>0.00</td>
      <td>0.00</td>
      <td>0.26</td>
    </tr>
    <tr>
      <th>Striatum ventral region</th>
      <td>1.00</td>
      <td>1.00</td>
      <td>1.00</td>
      <td>0.01</td>
      <td>0.00</td>
      <td>0.73</td>
    </tr>
    <tr>
      <th>Ventricular systems</th>
      <td>0.72</td>
      <td>0.13</td>
      <td>1.00</td>
      <td>0.40</td>
      <td>0.26</td>
      <td>0.97</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>mc.tl.communication_group_spatial(adata = adata,
                                  database_name = &#39;MetaChatDB&#39;,
                                  group_name = &quot;tissue_type&quot;,
                                  sum_metabolites = sum_metabolites,
                                  n_permutations = 100)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Computing group-level MCC for MetaChatDB-HMDB0000216
Computing group-level MCC for MetaChatDB-HMDB0000073
Computing group-level MCC for MetaChatDB-total-total
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Processing pairs: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 36/36 [08:56&lt;00:00, 14.90s/it]
</pre></div></div>
</div>
<div class="admonition important"><p class="admonition-title"><p>importan</p>
</p><p><p>This is a note.</p>
</p></div><p>Correspondingly, the group-level MCC results considering the spatial distance distribution are stored in <code class="docutils literal notranslate"><span class="pre">adata.uns['MetaChat_group_spatial-tissue_type-MetaChatDB-HMDB0000073']</span></code></p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>fig, ax = plt.subplots(1, 2, figsize=(11,4.5), subplot_kw={&#39;projection&#39;: &#39;polar&#39;})
mc.pl.plot_group_communication_chord(adata = adata,
                                     database_name = &#39;MetaChatDB&#39;,
                                     group_name = &quot;tissue_type&quot;,
                                     metabolite_name = &#39;HMDB0000073&#39;,
                                     permutation_spatial = False,
                                     p_value_cutoff = 1,
                                     space = 10,
                                     self_communication_off = False,
                                     ax = ax[0])
ax[0].set_title(&#39;With Self Communication&#39;, fontsize=14)

mc.pl.plot_group_communication_chord(adata = adata,
                                     database_name = &#39;MetaChatDB&#39;,
                                     group_name = &quot;tissue_type&quot;,
                                     metabolite_name = &#39;HMDB0000073&#39;,
                                     permutation_spatial = False,
                                     p_value_cutoff = 1,
                                     space = 10,
                                     self_communication_off = True,
                                     ax = ax[1])
ax[1].set_title(&#39;Without Self Communication&#39;, fontsize=14)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 1.0, &#39;Without Self Communication&#39;)
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_42_1.png" src="../_images/notebooks_Multi_omics_data_42_1.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>fig, ax = plt.subplots(1, 2, figsize=(11,3))
mc.pl.plot_group_communication_heatmap(adata = adata,
                                       database_name = &#39;MetaChatDB&#39;,
                                       group_name = &quot;tissue_type&quot;,
                                       metabolite_name = &#39;HMDB0000073&#39;,
                                       cmap = &#39;green&#39;,
                                       permutation_spatial = False,
                                       ax = ax[0])
ax[0].set_title(&#39;HMDB0000073&#39;, fontsize=14)

mc.pl.plot_group_communication_heatmap(adata = adata,
                                       database_name = &#39;MetaChatDB&#39;,
                                       group_name = &quot;tissue_type&quot;,
                                       metabolite_name = &#39;HMDB0000216&#39;,
                                       cmap = &#39;blue&#39;,
                                       permutation_spatial = False,
                                       ax = ax[1])
ax[1].set_title(&#39;HMDB0000216&#39;, fontsize=14)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 1.0, &#39;HMDB0000216&#39;)
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_43_1.png" src="../_images/notebooks_Multi_omics_data_43_1.png" />
</div>
</div>
</section>
<section id="MCC-remodelling-in-receiver-cells">
<h3>MCC remodelling in receiver cells<a class="headerlink" href="#MCC-remodelling-in-receiver-cells" title="Link to this heading"></a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>adata_raw = sc.read_h5ad(&#39;data/multiomics_mousebrain_parkinson_RNA.h5ad&#39;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>metabolite_name = &#39;HMDB0000073&#39;
df_deg, df_yhat = mc.tl.communication_responseGenes(adata = adata,
                                                    adata_raw = adata_raw,
                                                    database_name = &#39;MetaChatDB&#39;,
                                                    metabolite_name = metabolite_name,
                                                    summary = &#39;receiver&#39;)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
  |                                                  | 0 % ~calculating   |+                                                 | 1 % ~58m 04s       |++                                                | 2 % ~30m 13s       |++                                                | 3 % ~21m 09s       |+++                                               | 4 % ~16m 21s       |+++                                               | 5 % ~13m 31s       |++++                                              | 6 % ~11m 37s       |++++                                              | 7 % ~10m 20s       |+++++                                             | 8 % ~09m 23s       |+++++                                             | 9 % ~08m 31s       |++++++                                            | 10% ~07m 51s       |++++++                                            | 11% ~07m 17s       |+++++++                                           | 12% ~06m 49s       |+++++++                                           | 13% ~06m 25s       |++++++++                                          | 14% ~06m 05s       |++++++++                                          | 15% ~05m 50s       |+++++++++                                         | 16% ~05m 34s       |+++++++++                                         | 17% ~05m 20s       |++++++++++                                        | 18% ~05m 07s       |++++++++++                                        | 19% ~04m 54s       |+++++++++++                                       | 20% ~04m 43s       |+++++++++++                                       | 21% ~04m 32s       |++++++++++++                                      | 22% ~04m 22s       |++++++++++++                                      | 23% ~04m 16s       |+++++++++++++                                     | 24% ~04m 08s       |+++++++++++++                                     | 25% ~04m 01s       |++++++++++++++                                    | 26% ~03m 53s       |++++++++++++++                                    | 27% ~03m 47s       |+++++++++++++++                                   | 28% ~03m 40s       |+++++++++++++++                                   | 29% ~03m 34s       |++++++++++++++++                                  | 30% ~03m 31s       |++++++++++++++++                                  | 31% ~03m 25s       |+++++++++++++++++                                 | 32% ~03m 19s       |+++++++++++++++++                                 | 33% ~03m 13s       |++++++++++++++++++                                | 34% ~03m 09s       |++++++++++++++++++                                | 35% ~03m 04s       |+++++++++++++++++++                               | 36% ~02m 59s       |+++++++++++++++++++                               | 37% ~02m 54s       |++++++++++++++++++++                              | 38% ~02m 50s       |++++++++++++++++++++                              | 39% ~02m 46s       |+++++++++++++++++++++                             | 40% ~02m 41s       |+++++++++++++++++++++                             | 41% ~02m 39s       |++++++++++++++++++++++                            | 42% ~02m 35s       |++++++++++++++++++++++                            | 43% ~02m 31s       |+++++++++++++++++++++++                           | 44% ~02m 27s       |+++++++++++++++++++++++                           | 45% ~02m 23s       |++++++++++++++++++++++++                          | 46% ~02m 20s       |++++++++++++++++++++++++                          | 47% ~02m 16s       |+++++++++++++++++++++++++                         | 48% ~02m 13s       |+++++++++++++++++++++++++                         | 49% ~02m 09s       |++++++++++++++++++++++++++                        | 51% ~02m 06s       |++++++++++++++++++++++++++                        | 52% ~02m 03s       |+++++++++++++++++++++++++++                       | 53% ~01m 60s       |+++++++++++++++++++++++++++                       | 54% ~01m 57s       |++++++++++++++++++++++++++++                      | 55% ~01m 54s       |++++++++++++++++++++++++++++                      | 56% ~01m 51s       |+++++++++++++++++++++++++++++                     | 57% ~01m 48s       |+++++++++++++++++++++++++++++                     | 58% ~01m 45s       |++++++++++++++++++++++++++++++                    | 59% ~01m 42s       |++++++++++++++++++++++++++++++                    | 60% ~01m 39s       |+++++++++++++++++++++++++++++++                   | 61% ~01m 36s       |+++++++++++++++++++++++++++++++                   | 62% ~01m 33s       |++++++++++++++++++++++++++++++++                  | 63% ~01m 31s       |++++++++++++++++++++++++++++++++                  | 64% ~01m 28s       |+++++++++++++++++++++++++++++++++                 | 65% ~01m 25s       |+++++++++++++++++++++++++++++++++                 | 66% ~01m 22s       |++++++++++++++++++++++++++++++++++                | 67% ~01m 19s       |++++++++++++++++++++++++++++++++++                | 68% ~01m 17s       |+++++++++++++++++++++++++++++++++++               | 69% ~01m 14s       |+++++++++++++++++++++++++++++++++++               | 70% ~01m 11s       |++++++++++++++++++++++++++++++++++++              | 71% ~01m 09s       |++++++++++++++++++++++++++++++++++++              | 72% ~01m 07s       |+++++++++++++++++++++++++++++++++++++             | 73% ~01m 04s       |+++++++++++++++++++++++++++++++++++++             | 74% ~01m 02s       |++++++++++++++++++++++++++++++++++++++            | 75% ~60s           |++++++++++++++++++++++++++++++++++++++            | 76% ~57s           |+++++++++++++++++++++++++++++++++++++++           | 77% ~54s           |+++++++++++++++++++++++++++++++++++++++           | 78% ~52s           |++++++++++++++++++++++++++++++++++++++++          | 79% ~49s           |++++++++++++++++++++++++++++++++++++++++          | 80% ~47s           |+++++++++++++++++++++++++++++++++++++++++         | 81% ~44s           |+++++++++++++++++++++++++++++++++++++++++         | 82% ~42s           |++++++++++++++++++++++++++++++++++++++++++        | 83% ~39s           |++++++++++++++++++++++++++++++++++++++++++        | 84% ~37s           |+++++++++++++++++++++++++++++++++++++++++++       | 85% ~35s           |+++++++++++++++++++++++++++++++++++++++++++       | 86% ~32s           |++++++++++++++++++++++++++++++++++++++++++++      | 87% ~30s           |++++++++++++++++++++++++++++++++++++++++++++      | 88% ~28s           |+++++++++++++++++++++++++++++++++++++++++++++     | 89% ~25s           |+++++++++++++++++++++++++++++++++++++++++++++     | 90% ~23s           |++++++++++++++++++++++++++++++++++++++++++++++    | 91% ~21s           |++++++++++++++++++++++++++++++++++++++++++++++    | 92% ~18s           |+++++++++++++++++++++++++++++++++++++++++++++++   | 93% ~16s           |+++++++++++++++++++++++++++++++++++++++++++++++   | 94% ~14s           |++++++++++++++++++++++++++++++++++++++++++++++++  | 95% ~11s           |++++++++++++++++++++++++++++++++++++++++++++++++  | 96% ~09s           |+++++++++++++++++++++++++++++++++++++++++++++++++ | 97% ~07s           |+++++++++++++++++++++++++++++++++++++++++++++++++ | 98% ~05s           |++++++++++++++++++++++++++++++++++++++++++++++++++| 99% ~02s           |++++++++++++++++++++++++++++++++++++++++++++++++++| 100% elapsed=03m 42s
2 parameter combinations, 2 use sequential method, 2 use subsampling method
Running Clustering on Parameter Combinations...
done.
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>import pickle
deg_result = {&quot;df_deg&quot;: df_deg, &quot;df_yhat&quot;: df_yhat}
with open(&quot;results/responseGenes_&quot; + metabolite_name + &quot;.pkl&quot;, &#39;wb&#39;) as handle:
    pickle.dump(deg_result, handle, protocol=pickle.HIGHEST_PROTOCOL)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>metabolite_name = &#39;HMDB0000073&#39;
import pickle
with open(&quot;results/responseGenes_&quot; + metabolite_name + &quot;.pkl&quot;, &#39;rb&#39;) as file:
    deg_result = pickle.load(file)
df_deg = deg_result[&quot;df_deg&quot;]
df_yhat = deg_result[&quot;df_yhat&quot;]
df_deg_clus, df_yhat_clus = mc.tl.communication_responseGenes_cluster(df_deg.iloc[:,:3], df_yhat, deg_clustering_res=0.2)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>mc.pl.plot_communication_responseGenes(df_deg = df_deg_clus,
                                       df_yhat = df_yhat_clus,
                                       top_ngene_per_cluster = 8,
                                       colormap = &#39;viridis&#39;,
                                       color_range = (-2,2),
                                       font_scale = 1,
                                       figsize = (10,6),
                                       return_genes = False)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_49_0.png" src="../_images/notebooks_Multi_omics_data_49_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>df_result = mc.tl.communication_responseGenes_keggEnrich(gene_list = df_deg[df_deg[&#39;pvalue&#39;] &lt; 0.05].index.tolist()[:1000],
                                                         gene_sets = &quot;KEGG_2019_Mouse&quot;,
                                                         organism = &quot;Mouse&quot;)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>df_result.to_csv(&quot;results/responseGenes_keggEnrich_&quot; + metabolite_name + &quot;.csv&quot;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-none notranslate"><div class="highlight"><pre><span></span>mc.pl.plot_communication_responseGenes_keggEnrich(df_result = df_result,
                                                  show_term_order = [1,2,6,7,9],
                                                  cmap = &#39;blue&#39;,
                                                  maxshow_gene = 10,
                                                  figsize = (7,5))
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_Multi_omics_data_52_0.png" src="../_images/notebooks_Multi_omics_data_52_0.png" />
</div>
</div>
</section>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="toy_example1.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">Toy example 1</div>
              </div>
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="../tutorials.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">&lt;no title&gt;</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2024, Songhao Luo
            </div>
            Made with 
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            <div class="icons">
              <a class="muted-link fab fa-github" href="https://github.com/SonghaoLuo/metachat" aria-label="GitHub"></a>
              
            </div>
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Example of spatial multi-omics data (mouse brain with Parkinson’s disease)</a><ul>
<li><a class="reference internal" href="#Data-preprocessing">Data preprocessing</a><ul>
<li><a class="reference internal" href="#Preprocessing-for-transcriptomics-data">Preprocessing for transcriptomics data</a></li>
<li><a class="reference internal" href="#Preprocessing-for-metabolomics-data">Preprocessing for metabolomics data</a></li>
<li><a class="reference internal" href="#Combine-the-transcriptomics-and-metabolomics-data">Combine the transcriptomics and metabolomics data</a></li>
</ul>
</li>
<li><a class="reference internal" href="#MCC-Inference">MCC Inference</a><ul>
<li><a class="reference internal" href="#Import-MetaChatDB">Import MetaChatDB</a></li>
<li><a class="reference internal" href="#Identify-long-range-channels">Identify long-range channels</a></li>
</ul>
</li>
<li><a class="reference internal" href="#Downstream-analysis">Downstream analysis</a><ul>
<li><a class="reference internal" href="#Summary-communication-among-different-scales">Summary communication among different scales</a></li>
<li><a class="reference internal" href="#Identifing-MCC-flow">Identifing MCC flow</a></li>
<li><a class="reference internal" href="#Constructing-group-level-MCC">Constructing group-level MCC</a></li>
<li><a class="reference internal" href="#MCC-remodelling-in-receiver-cells">MCC remodelling in receiver cells</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="../_static/jquery.js?v=5d32c60e"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
    <script src="../_static/documentation_options.js?v=e3a6060d"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/scripts/furo.js?v=4e2eecee"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    </body>
</html>