<!doctype html>
<html class="no-js" lang="en" data-content_root="../../">
  <head><meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="../../genindex.html"><link rel="search" title="Search" href="../../search.html"><link rel="next" title="Downstream analysis on spatial multi-omics data of mouse brain with Parkinson’s disease" href="1.3_mouse_brain_parkinson_downstream.html"><link rel="prev" title="MetaChat analysis on spatial multi-omics data of mouse brain with Parkinson’s disease" href="1.1_mouse_brain_parkinson_metachat.html">
        <link rel="prefetch" href="../../_static/logo.png" as="image">

    <!-- Generated with Sphinx 7.4.7 and Furo 2025.09.25 -->
        <title>Comparative analysis on spatial multi-omics data of mouse brain with Parkinson’s disease - MetaChat 0.0.5 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=d111a655" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/furo.css?v=580074bf" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/sphinx_rtd_size.css?v=8f58cd29" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css?v=6644e6bb" />
    <link rel="stylesheet" type="text/css" href="../../_static/sg_gallery.css?v=d2d258e8" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/furo-extensions.css?v=8dab3a3b" />
    <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/override.css?v=46f5334b" />
    
    


<style>
  body {
    --color-code-background: #f2f2f2;
  --color-code-foreground: #1e1e1e;
  --color-brand-primary: #357473;
  --color-brand-content: #357473;
  --admonition-font-size: var(--font-size-normal);
  --admonition-title-font-size: var(--font-size-normal);
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-with-moon" viewBox="0 0 24 24">
    <title>Auto light/dark, in light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path style="opacity: 50%" d="M 5.411 14.504 C 5.471 14.504 5.532 14.504 5.591 14.504 C 3.639 16.319 4.383 19.569 6.931 20.352 C 7.693 20.586 8.512 20.551 9.25 20.252 C 8.023 23.207 4.056 23.725 2.11 21.184 C 0.166 18.642 1.702 14.949 4.874 14.536 C 5.051 14.512 5.231 14.5 5.411 14.5 L 5.411 14.504 Z"/>
      <line x1="14.5" y1="3.25" x2="14.5" y2="1.25"/>
      <line x1="14.5" y1="15.85" x2="14.5" y2="17.85"/>
      <line x1="10.044" y1="5.094" x2="8.63" y2="3.68"/>
      <line x1="19" y1="14.05" x2="20.414" y2="15.464"/>
      <line x1="8.2" y1="9.55" x2="6.2" y2="9.55"/>
      <line x1="20.8" y1="9.55" x2="22.8" y2="9.55"/>
      <line x1="10.044" y1="14.006" x2="8.63" y2="15.42"/>
      <line x1="19" y1="5.05" x2="20.414" y2="3.636"/>
      <circle cx="14.5" cy="9.55" r="3.6"/>
    </svg>
  </symbol>
  <symbol id="svg-moon-with-sun" viewBox="0 0 24 24">
    <title>Auto light/dark, in dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path d="M 8.282 7.007 C 8.385 7.007 8.494 7.007 8.595 7.007 C 5.18 10.184 6.481 15.869 10.942 17.24 C 12.275 17.648 13.706 17.589 15 17.066 C 12.851 22.236 5.91 23.143 2.505 18.696 C -0.897 14.249 1.791 7.786 7.342 7.063 C 7.652 7.021 7.965 7 8.282 7 L 8.282 7.007 Z"/>
      <line style="opacity: 50%" x1="18" y1="3.705" x2="18" y2="2.5"/>
      <line style="opacity: 50%" x1="18" y1="11.295" x2="18" y2="12.5"/>
      <line style="opacity: 50%" x1="15.316" y1="4.816" x2="14.464" y2="3.964"/>
      <line style="opacity: 50%" x1="20.711" y1="10.212" x2="21.563" y2="11.063"/>
      <line style="opacity: 50%" x1="14.205" y1="7.5" x2="13.001" y2="7.5"/>
      <line style="opacity: 50%" x1="21.795" y1="7.5" x2="23" y2="7.5"/>
      <line style="opacity: 50%" x1="15.316" y1="10.184" x2="14.464" y2="11.036"/>
      <line style="opacity: 50%" x1="20.711" y1="4.789" x2="21.563" y2="3.937"/>
      <circle style="opacity: 50%" cx="18" cy="7.5" r="2.169"/>
    </svg>
  </symbol>
  <symbol id="svg-pencil" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-pencil-code">
      <path d="M4 20h4l10.5 -10.5a2.828 2.828 0 1 0 -4 -4l-10.5 10.5v4" />
      <path d="M13.5 6.5l4 4" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
  <symbol id="svg-eye" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-eye-code">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0" />
      <path
        d="M11.11 17.958c-3.209 -.307 -5.91 -2.293 -8.11 -5.958c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6c-.21 .352 -.427 .688 -.647 1.008" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle site navigation sidebar">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc" aria-label="Toggle table of contents sidebar">
<label class="overlay sidebar-overlay" for="__navigation"></label>
<label class="overlay toc-overlay" for="__toc"></label>

<a class="skip-to-content muted-link" href="#furo-main-content">Skip to content</a>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <span class="icon"><svg><use href="#svg-menu"></use></svg></span>
      </label>
    </div>
    <div class="header-center">
      <a href="../../index.html"><div class="brand">MetaChat 0.0.5 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle" aria-label="Toggle Light / Dark / Auto color theme">
          <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
          <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <span class="icon"><svg><use href="#svg-toc"></use></svg></span>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../../index.html">
  <div class="sidebar-logo-container">
    <img class="sidebar-logo" src="../../_static/logo.png" alt="Logo"/>
  </div>
  
  
</a><form class="sidebar-search-container" method="get" action="../../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../installation.html">Installation</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../basic/index.html">Basic tutorials</a><input aria-label="Toggle navigation of Basic tutorials" class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><span class="icon"><svg><use href="#svg-arrow-right"></use></svg></span></label><ul>
<li class="toctree-l2 has-children"><a class="reference internal" href="../../basic/identify_anatomical_features/index.html">Identifying Anatomical Features</a><input aria-label="Toggle navigation of Identifying Anatomical Features" class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><span class="icon"><svg><use href="#svg-arrow-right"></use></svg></span></label><ul>
<li class="toctree-l3"><a class="reference internal" href="../../basic/identify_anatomical_features/identifyLRC.html">Identifying Long-Range Channels (LRCs)</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../basic/identify_anatomical_features/identifyBarrier.html">Annotating barrier condition</a></li>
</ul>
</li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../../basic/estimate_metabolomics/index.html">Estimating Spatial Metabolomics from Spatial Transcriptomics</a><input aria-label="Toggle navigation of Estimating Spatial Metabolomics from Spatial Transcriptomics" class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" role="switch" type="checkbox"/><label for="toctree-checkbox-3"><span class="icon"><svg><use href="#svg-arrow-right"></use></svg></span></label><ul>
<li class="toctree-l3"><a class="reference internal" href="../../basic/estimate_metabolomics/scFEA.html">using scFEA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../basic/estimate_metabolomics/compass.html">using COMPASS</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../basic/estimate_metabolomics/mebocost.html">using MEBOCOST</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../basic/tips.html">Tip for Analyzing Multiple Slides or Biological Replicates</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../basic/function_guideline.html">Guideline for Function and Visualization</a></li>
</ul>
</li>
<li class="toctree-l1 current has-children"><a class="reference internal" href="../index.html">Real dataset tutorials</a><input aria-label="Toggle navigation of Real dataset tutorials" checked="" class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" role="switch" type="checkbox"/><label for="toctree-checkbox-4"><span class="icon"><svg><use href="#svg-arrow-right"></use></svg></span></label><ul class="current">
<li class="toctree-l2 current has-children"><a class="reference internal" href="index.html">Spatial Multi-Omics Data from Mouse Brain with Parkinson’s Disease</a><input aria-label="Toggle navigation of Spatial Multi-Omics Data from Mouse Brain with Parkinson’s Disease" checked="" class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" role="switch" type="checkbox"/><label for="toctree-checkbox-5"><span class="icon"><svg><use href="#svg-arrow-right"></use></svg></span></label><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="1.1_mouse_brain_parkinson_metachat.html">MetaChat analysis</a></li>
<li class="toctree-l3 current current-page"><a class="current reference internal" href="#">Comparative analysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="1.3_mouse_brain_parkinson_downstream.html">Downstream analysis</a></li>
</ul>
</li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../2_human_gastric_cancer/index.html">Spatial Multi-Omics Data from Human Gastric Cancer</a><input aria-label="Toggle navigation of Spatial Multi-Omics Data from Human Gastric Cancer" class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" role="switch" type="checkbox"/><label for="toctree-checkbox-6"><span class="icon"><svg><use href="#svg-arrow-right"></use></svg></span></label><ul>
<li class="toctree-l3"><a class="reference internal" href="../2_human_gastric_cancer/2.1_human_gastric_cancer_metachat.html">MetaChat analysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../2_human_gastric_cancer/2.2_human_gastric_cancer_downstream.html">Downstream analysis</a></li>
</ul>
</li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../3_mouse_small_intestine/index.html">Visium HD Dataset from Mouse Small Intestine</a><input aria-label="Toggle navigation of Visium HD Dataset from Mouse Small Intestine" class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" role="switch" type="checkbox"/><label for="toctree-checkbox-7"><span class="icon"><svg><use href="#svg-arrow-right"></use></svg></span></label><ul>
<li class="toctree-l3"><a class="reference internal" href="../3_mouse_small_intestine/3.1_mouse_small_intestine_metachat.html">MetaChat analysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../3_mouse_small_intestine/3.2_mouse_small_intestine_downstream.html">Downstream analysis</a></li>
</ul>
</li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../4_mouse_brain_3D/index.html">3D Transcriptomics Dataset from Whole Mouse Brain</a><input aria-label="Toggle navigation of 3D Transcriptomics Dataset from Whole Mouse Brain" class="toctree-checkbox" id="toctree-checkbox-8" name="toctree-checkbox-8" role="switch" type="checkbox"/><label for="toctree-checkbox-8"><span class="icon"><svg><use href="#svg-arrow-right"></use></svg></span></label><ul>
<li class="toctree-l3"><a class="reference internal" href="../4_mouse_brain_3D/4.1_mouse_brain_3D_metachat.html">MetaChat analysis and Downstream analysis</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../api/index.html">API</a><input aria-label="Toggle navigation of API" class="toctree-checkbox" id="toctree-checkbox-9" name="toctree-checkbox-9" role="switch" type="checkbox"/><label for="toctree-checkbox-9"><span class="icon"><svg><use href="#svg-arrow-right"></use></svg></span></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pp.MetaChatDB.html">metachat.pp.MetaChatDB</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pp.generate_adata_met_compass.html">metachat.pp.generate_adata_met_compass</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pp.generate_adata_met_scFEA.html">metachat.pp.generate_adata_met_scFEA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pp.generate_adata_met_mebocost.html">metachat.pp.generate_adata_met_mebocost</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pp.global_intensity_scaling.html">metachat.pp.global_intensity_scaling</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pp.load_barrier_segments.html">metachat.pp.load_barrier_segments</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pp.LRC_unfiltered.html">metachat.pp.LRC_unfiltered</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pp.LRC_cluster.html">metachat.pp.LRC_cluster</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pp.LRC_filtered.html">metachat.pp.LRC_filtered</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pp.compute_costDistance.html">metachat.pp.compute_costDistance</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.tl.metabolic_communication.html">metachat.tl.metabolic_communication</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.tl.summary_communication.html">metachat.tl.summary_communication</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.tl.communication_flow.html">metachat.tl.communication_flow</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.tl.communication_group.html">metachat.tl.communication_group</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.tl.communication_group_spatial.html">metachat.tl.communication_group_spatial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.tl.summary_pathway.html">metachat.tl.summary_pathway</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.tl.communication_responseGenes.html">metachat.tl.communication_responseGenes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.tl.communication_responseGenes_cluster.html">metachat.tl.communication_responseGenes_cluster</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.tl.communication_responseGenes_keggEnrich.html">metachat.tl.communication_responseGenes_keggEnrich</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.tl.compute_direction_histogram_per_pair.html">metachat.tl.compute_direction_histogram_per_pair</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_communication_flow.html">metachat.pl.plot_communication_flow</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_group_communication_chord.html">metachat.pl.plot_group_communication_chord</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_group_communication_heatmap.html">metachat.pl.plot_group_communication_heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_group_communication_compare_hierarchy_diagram.html">metachat.pl.plot_group_communication_compare_hierarchy_diagram</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_MSpair_contribute_group.html">metachat.pl.plot_MSpair_contribute_group</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_summary_pathway.html">metachat.pl.plot_summary_pathway</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_metapathway_pair_contribution_bubbleplot.html">metachat.pl.plot_metapathway_pair_contribution_bubbleplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_communication_responseGenes.html">metachat.pl.plot_communication_responseGenes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_communication_responseGenes_keggEnrich.html">metachat.pl.plot_communication_responseGenes_keggEnrich</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_DEG_volcano.html">metachat.pl.plot_DEG_volcano</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_3d_feature.html">metachat.pl.plot_3d_feature</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_3d_LRC_with_two_slices.html">metachat.pl.plot_3d_LRC_with_two_slices</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_dis_thr.html">metachat.pl.plot_dis_thr</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_LRC_markers.html">metachat.pl.plot_LRC_markers</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_spot_distance.html">metachat.pl.plot_spot_distance</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_graph_connectivity.html">metachat.pl.plot_graph_connectivity</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../api/metachat.pl.plot_direction_similarity.html">metachat.pl.plot_direction_similarity</a></li>
</ul>
</li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          

<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle" aria-label="Toggle Light / Dark / Auto color theme">
              <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
              <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <span class="icon"><svg><use href="#svg-toc"></use></svg></span>
          </label>
        </div>
        <article role="main" id="furo-main-content">
          <section id="comparative-analysis-on-spatial-multi-omics-data-of-mouse-brain-with-parkinson-s-disease">
<h1>Comparative analysis on spatial multi-omics data of mouse brain with Parkinson’s disease<a class="headerlink" href="#comparative-analysis-on-spatial-multi-omics-data-of-mouse-brain-with-parkinson-s-disease" title="Link to this heading"></a></h1>
<p>In this tutorial, we perform a comparative analysis of spatial multi-omics data from the mouse brain with Parkinson’s disease, where the left and right hemispheres represent the intact and lesioned conditions, respectively. This comparison allows us to investigate how MCC differs between diseased and healthy regions.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Importing packages</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">os</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">metachat</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mc</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">scanpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">sc</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">squidpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">sq</span>

<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">matplotlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">rcParams</span>
<span class="n">rcParams</span><span class="p">[</span><span class="s1">&#39;pdf.fonttype&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mi">42</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">warnings</span>
<span class="n">warnings</span><span class="o">.</span><span class="n">filterwarnings</span><span class="p">(</span><span class="s2">&quot;ignore&quot;</span><span class="p">,</span> <span class="n">category</span><span class="o">=</span><span class="ne">FutureWarning</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Setting your work dictionary</span>
<span class="n">os</span><span class="o">.</span><span class="n">chdir</span><span class="p">(</span><span class="s2">&quot;/home/project/metachat_packages/&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<section id="import-metachat-results">
<h2>Import MetaChat results<a class="headerlink" href="#import-metachat-results" title="Link to this heading"></a></h2>
<p>We manually annotated the MetaChat result using Napari, dividing the left hemisphere as “Intact” and the right hemisphere as “Lesion”. For convenience, we will directly import the annotated h5ad.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s1">&#39;datasets/mouse_brain_parkinson/metachat_result_divided.h5ad&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>The annotations are stored in adata.obs[‘lesion_shapes’], which can be visualized for further comparison.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span> <span class="o">=</span> <span class="p">(</span><span class="mi">6</span><span class="p">,</span><span class="mi">5</span><span class="p">))</span>
<span class="n">sq</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">spatial_scatter</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span>
                      <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;lesion_shapes&#39;</span><span class="p">,</span>
                      <span class="n">ax</span> <span class="o">=</span> <span class="n">ax</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_box_aspect</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/0d4824e3173433376780e4a8309cd5714db9b2a8cc8ae9cabafccccbc24710c1.png" src="../../_images/0d4824e3173433376780e4a8309cd5714db9b2a8cc8ae9cabafccccbc24710c1.png" />
</div>
</div>
</section>
<section id="clustering-through-mcc-signals">
<h2>Clustering through MCC signals<a class="headerlink" href="#clustering-through-mcc-signals" title="Link to this heading"></a></h2>
<p>We performed clustering based on MCC signals to investigate whether these communication activities show spatially organized or region-specific distributions within the tissue.</p>
<p>Therefore, we constructed a new <code class="docutils literal notranslate"><span class="pre">adata</span></code> object in which the features represent only MCC-related signals — including the total sender and receiver activities for each cell (spot).</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">MCC_matrix_sender</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;MetaChat-MetaChatDB-sum-sender&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
<span class="n">MCC_matrix_receiver</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;MetaChat-MetaChatDB-sum-receiver&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
<span class="n">MCC_matrix</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">MCC_matrix_sender</span><span class="p">,</span> <span class="n">MCC_matrix_receiver</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">adata_MCC</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">AnnData</span><span class="p">(</span><span class="n">MCC_matrix</span><span class="p">)</span>
<span class="n">adata_MCC</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span>
<span class="n">adata_MCC</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_MCC</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AnnData object with n_obs × n_vars = 3054 × 1034
    uns: &#39;spatial&#39;
    obsm: &#39;spatial&#39;
</pre></div>
</div>
</div>
</div>
<p>We next computed the spatial neighborhood graph and assessed the spatial autocorrelation of each MCC feature using Moran’s I statistic.
This analysis aims to evaluate whether MCC signals display spatial dependency — that is, whether cells with similar levels of metabolite-mediated communication tend to cluster together in the tissue.
Identifying spatially autocorrelated MCC features helps reveal region-specific communication patterns and highlights spatial domains with coordinated metabolic activity.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sq</span><span class="o">.</span><span class="n">gr</span><span class="o">.</span><span class="n">spatial_neighbors</span><span class="p">(</span><span class="n">adata_MCC</span><span class="p">)</span>
<span class="n">sq</span><span class="o">.</span><span class="n">gr</span><span class="o">.</span><span class="n">spatial_autocorr</span><span class="p">(</span>
    <span class="n">adata_MCC</span><span class="p">,</span>
    <span class="n">mode</span> <span class="o">=</span> <span class="s2">&quot;moran&quot;</span><span class="p">,</span>
    <span class="n">n_perms</span> <span class="o">=</span> <span class="mi">100</span><span class="p">,</span>
    <span class="n">n_jobs</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/opt/conda/envs/metachat_test_20251015/lib/python3.9/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
100%|██████████| 100/100 [00:03&lt;00:00, 31.50/s]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_MCC</span><span class="o">.</span><span class="n">var_names_make_unique</span><span class="p">()</span>
<span class="n">moranI</span> <span class="o">=</span> <span class="n">adata_MCC</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s2">&quot;moranI&quot;</span><span class="p">]</span>
<span class="n">adata_MCC_cluster</span> <span class="o">=</span> <span class="n">adata_MCC</span><span class="p">[:,</span> <span class="n">moranI</span><span class="o">.</span><span class="n">index</span><span class="p">[:</span><span class="mi">200</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<p>Similar to scRNA-seq analysis, we performed dimensionality reduction and clustering on the MCC-derived features to identify spatial domains with distinct metabolic communication profiles.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">random_seed</span> <span class="o">=</span> <span class="mi">20</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">highly_variable_genes</span><span class="p">(</span><span class="n">adata_MCC_cluster</span><span class="p">,</span> <span class="n">min_mean</span><span class="o">=</span><span class="mf">0.0125</span><span class="p">,</span> <span class="n">min_disp</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>
<span class="n">adata_MCC_cluster</span> <span class="o">=</span> <span class="n">adata_MCC_cluster</span><span class="p">[:,</span> <span class="n">adata_MCC_cluster</span><span class="o">.</span><span class="n">var</span><span class="o">.</span><span class="n">highly_variable</span><span class="p">]</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">pca</span><span class="p">(</span><span class="n">adata_MCC_cluster</span><span class="p">,</span> <span class="n">svd_solver</span><span class="o">=</span><span class="s1">&#39;arpack&#39;</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="n">random_seed</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">neighbors</span><span class="p">(</span><span class="n">adata_MCC_cluster</span><span class="p">,</span> <span class="n">n_neighbors</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">n_pcs</span><span class="o">=</span><span class="mi">40</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="n">random_seed</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">umap</span><span class="p">(</span><span class="n">adata_MCC_cluster</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="n">random_seed</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">leiden</span><span class="p">(</span><span class="n">adata_MCC_cluster</span><span class="p">,</span> <span class="n">resolution</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="n">random_seed</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span> <span class="o">=</span> <span class="p">(</span><span class="mi">6</span><span class="p">,</span><span class="mi">5</span><span class="p">))</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">spatial</span><span class="p">(</span><span class="n">adata_MCC_cluster</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;leiden&#39;</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Accent&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_box_aspect</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/264a78f77870ab748ed790e3abf513cdb95cb0feb532ad5a56bb8c61fb697111.png" src="../../_images/264a78f77870ab748ed790e3abf513cdb95cb0feb532ad5a56bb8c61fb697111.png" />
</div>
</div>
<p>Using MCC-derived features, distinct anatomical regions can still be clearly delineated, demonstrating that metabolic communication patterns reflect the underlying tissue organization.
Interestingly, the area originally clustered as the Striatum dorsal region based on gene expression is now divided into two subgroups when clustered by MCC signals, suggesting notable differences in metabolic communication between the intact and lesioned hemispheres.
In the following section, we perform differential analysis to further characterize these MCC differences.</p>
</section>
<section id="differential-mcc-among-each-tissue-type">
<h2>Differential MCC among each tissue type<a class="headerlink" href="#differential-mcc-among-each-tissue-type" title="Link to this heading"></a></h2>
<p>In this step, we perform differential analysis of MCC features across tissue types to detect metabolites and sensors exhibiting significant differences in communication activity, thereby revealing region-specific metabolic communication patterns.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_MCC</span><span class="o">.</span><span class="n">var_names_make_unique</span><span class="p">()</span>
<span class="n">moranI</span> <span class="o">=</span> <span class="n">adata_MCC</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s2">&quot;moranI&quot;</span><span class="p">]</span>
<span class="n">adata_MCC_key</span> <span class="o">=</span> <span class="n">adata_MCC</span><span class="p">[:,</span> <span class="n">moranI</span><span class="p">[</span><span class="s1">&#39;pval_norm&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mf">0.05</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">adata_MCC_key</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;tissue_type&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;tissue_type&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">rank_genes_groups</span><span class="p">(</span><span class="n">adata_MCC_key</span><span class="p">,</span> <span class="n">groupby</span><span class="o">=</span><span class="s2">&quot;tissue_type&quot;</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s2">&quot;t-test_overestim_var&quot;</span><span class="p">)</span>
<span class="n">DEG_result</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">get</span><span class="o">.</span><span class="n">rank_genes_groups_df</span><span class="p">(</span><span class="n">adata_MCC_key</span><span class="p">,</span> <span class="n">group</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">pval_cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span> <span class="n">log2fc_min</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dp</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">rank_genes_groups_dotplot</span><span class="p">(</span>
    <span class="n">adata_MCC_key</span><span class="p">,</span>
    <span class="n">n_genes</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
    <span class="n">standard_scale</span><span class="o">=</span><span class="s1">&#39;var&#39;</span><span class="p">,</span>
    <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;RdBu_r&#39;</span><span class="p">,</span>
    <span class="n">swap_axes</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
    <span class="n">show</span><span class="o">=</span><span class="kc">False</span>
<span class="p">)</span>

<span class="k">for</span> <span class="n">ax</span> <span class="ow">in</span> <span class="n">plt</span><span class="o">.</span><span class="n">gcf</span><span class="p">()</span><span class="o">.</span><span class="n">axes</span><span class="p">:</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;x&#39;</span><span class="p">,</span> <span class="n">labelrotation</span><span class="o">=</span><span class="mi">45</span><span class="p">)</span>
    <span class="k">for</span> <span class="n">tick</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">():</span>
        <span class="n">tick</span><span class="o">.</span><span class="n">set_horizontalalignment</span><span class="p">(</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>WARNING: dendrogram data not found (using key=dendrogram_tissue_type). Running `sc.tl.dendrogram` with default parameters. For fine tuning it is recommended to run `sc.tl.dendrogram` independently.
WARNING: You’re trying to run this on 668 dimensions of `.X`, if you really want this, set `use_rep=&#39;X&#39;`.
         Falling back to preprocessing with `sc.pp.pca` and default params.
</pre></div>
</div>
<img alt="../../_images/acde6d6df4bcc73fbd9d9a91c383c3c0341885d5fe8956020f7a16bbb09976e3.png" src="../../_images/acde6d6df4bcc73fbd9d9a91c383c3c0341885d5fe8956020f7a16bbb09976e3.png" />
</div>
</div>
</section>
<section id="differential-mcc-between-intact-and-lesion-region">
<h2>Differential MCC between intact and lesion region<a class="headerlink" href="#differential-mcc-between-intact-and-lesion-region" title="Link to this heading"></a></h2>
<p>To focus on the most spatially structured communication features, we selected the top 100 sender-related and top 100 receiver-related MCC signals with the highest Moran’s I scores.
These features represent metabolite–sensor activities that exhibit the strongest spatial autocorrelation, suggesting they are likely to be biologically organized rather than random.
The selected features were then combined for downstream differential analysis between the Intact and Lesion hemispheres.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_MCC_sender</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">AnnData</span><span class="p">(</span><span class="n">MCC_matrix_sender</span><span class="p">)</span>
<span class="n">adata_MCC_sender</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span>
<span class="n">adata_MCC_sender</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sq</span><span class="o">.</span><span class="n">gr</span><span class="o">.</span><span class="n">spatial_neighbors</span><span class="p">(</span><span class="n">adata_MCC_sender</span><span class="p">)</span>
<span class="n">sq</span><span class="o">.</span><span class="n">gr</span><span class="o">.</span><span class="n">spatial_autocorr</span><span class="p">(</span>
    <span class="n">adata_MCC_sender</span><span class="p">,</span>
    <span class="n">mode</span> <span class="o">=</span> <span class="s2">&quot;moran&quot;</span><span class="p">,</span>
    <span class="n">n_perms</span> <span class="o">=</span> <span class="mi">100</span><span class="p">,</span>
    <span class="n">n_jobs</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>  0%|          | 0/100 [00:00&lt;?, ?/s]
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>100%|██████████| 100/100 [00:02&lt;00:00, 36.74/s]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_MCC_receiver</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">AnnData</span><span class="p">(</span><span class="n">MCC_matrix_receiver</span><span class="p">)</span>
<span class="n">adata_MCC_receiver</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span>
<span class="n">adata_MCC_receiver</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sq</span><span class="o">.</span><span class="n">gr</span><span class="o">.</span><span class="n">spatial_neighbors</span><span class="p">(</span><span class="n">adata_MCC_receiver</span><span class="p">)</span>
<span class="n">sq</span><span class="o">.</span><span class="n">gr</span><span class="o">.</span><span class="n">spatial_autocorr</span><span class="p">(</span>
    <span class="n">adata_MCC_receiver</span><span class="p">,</span>
    <span class="n">mode</span> <span class="o">=</span> <span class="s2">&quot;moran&quot;</span><span class="p">,</span>
    <span class="n">n_perms</span> <span class="o">=</span> <span class="mi">100</span><span class="p">,</span>
    <span class="n">n_jobs</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>  0%|          | 0/100 [00:00&lt;?, ?/s]
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>100%|██████████| 100/100 [00:02&lt;00:00, 37.93/s]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_MCC</span><span class="o">.</span><span class="n">var_names_make_unique</span><span class="p">()</span>
<span class="n">selected_vars</span> <span class="o">=</span> <span class="n">adata_MCC_sender</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s2">&quot;moranI&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">[:</span><span class="mi">100</span><span class="p">]</span><span class="o">.</span><span class="n">union</span><span class="p">(</span>
    <span class="n">adata_MCC_receiver</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s2">&quot;moranI&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">[:</span><span class="mi">100</span><span class="p">]</span>
<span class="p">)</span>
<span class="n">adata_MCC_diff</span> <span class="o">=</span> <span class="n">adata_MCC</span><span class="p">[:,</span> <span class="n">selected_vars</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_MCC_diff</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AnnData object with n_obs × n_vars = 3054 × 200
    uns: &#39;spatial&#39;, &#39;spatial_neighbors&#39;, &#39;moranI&#39;
    obsm: &#39;spatial&#39;
    obsp: &#39;spatial_connectivities&#39;, &#39;spatial_distances&#39;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_MCC_diff</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;lesion_shapes&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;lesion_shapes&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">adata_MCC_diff</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;lesion_shapes_colors&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;lesion_shapes_colors&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">adata_MCC_diff</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;tissue_type&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;tissue_type&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_MCC_diff</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;disease_group&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>
<span class="n">adata_MCC_diff</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">adata_MCC_diff</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;lesion_shapes&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="kc">False</span><span class="p">,</span> <span class="s1">&#39;disease_group&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;Intact&#39;</span>
<span class="n">adata_MCC_diff</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">adata_MCC_diff</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;lesion_shapes&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="kc">True</span><span class="p">,</span> <span class="s1">&#39;disease_group&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;Lesion&#39;</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/tmp/ipykernel_3734810/1412599778.py:2: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value &#39;Intact&#39; has dtype incompatible with int64, please explicitly cast to a compatible dtype first.
  adata_MCC_diff.obs.loc[adata_MCC_diff.obs[&#39;lesion_shapes&#39;] == False, &#39;disease_group&#39;] = &#39;Intact&#39;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">rank_genes_groups</span><span class="p">(</span><span class="n">adata_MCC_diff</span><span class="p">,</span> <span class="n">groupby</span><span class="o">=</span><span class="s2">&quot;disease_group&quot;</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s2">&quot;t-test_overestim_var&quot;</span><span class="p">)</span>
<span class="n">DEG_result</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">get</span><span class="o">.</span><span class="n">rank_genes_groups_df</span><span class="p">(</span><span class="n">adata_MCC_diff</span><span class="p">,</span> <span class="n">group</span><span class="o">=</span><span class="s2">&quot;Intact&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>We then visualized the differential MCC features using a volcano plot, which highlights metabolite–sensor pairs with significant differences in communication strength between the Intact and Lesion hemispheres.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_DEG_volcano</span><span class="p">(</span><span class="n">DEG_result</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;Axes: title={&#39;center&#39;: &#39;Volcano Plot&#39;}, xlabel=&#39;Log2 Fold Change&#39;, ylabel=&#39;-Log10 p-value&#39;&gt;
</pre></div>
</div>
<img alt="../../_images/f6f64b30a9d5c42cbbbf3c43260eb3288f5fcb36d0c92676d4765b1350019d83.png" src="../../_images/f6f64b30a9d5c42cbbbf3c43260eb3288f5fcb36d0c92676d4765b1350019d83.png" />
</div>
</div>
<p>Since a few metabolite–sensor pairs showed abnormally large fold changes, likely due to low expression levels or sparse spot coverage, we restricted the analysis to pairs with |log₂ fold change| &lt; 4 to avoid artifacts and focus on biologically meaningful differences.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">DEG_result_filtered</span> <span class="o">=</span> <span class="n">DEG_result</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="nb">abs</span><span class="p">(</span><span class="n">DEG_result</span><span class="p">[</span><span class="s1">&#39;logfoldchanges&#39;</span><span class="p">])</span><span class="o">&lt;=</span><span class="mi">4</span><span class="p">,:]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_DEG_volcano</span><span class="p">(</span><span class="n">DEG_result_filtered</span><span class="p">,</span>
                       <span class="n">neglog10_cap</span> <span class="o">=</span> <span class="mi">50</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>0 [-0.67501402  0.42160331]
1 [-0.40538805 -0.16830784]
</pre></div>
</div>
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;Axes: title={&#39;center&#39;: &#39;Volcano Plot&#39;}, xlabel=&#39;Log2 Fold Change&#39;, ylabel=&#39;-Log10 p-value&#39;&gt;
</pre></div>
</div>
<img alt="../../_images/ade0cb4db427e7849a028cb0c2fb146a519d1095d67c1bee18e977f888e770b4.png" src="../../_images/ade0cb4db427e7849a028cb0c2fb146a519d1095d67c1bee18e977f888e770b4.png" />
</div>
</div>
<p>We observed that dopamine (HMDB0000073) and norepinephrine (HMDB0000216) exhibited significant differences in MCC activity between the intact and lesioned hemispheres, which is consistent with previous reports on their roles in Parkinson’s disease.
In the following section, we perform further analyses focusing on MCCs associated with these two metabolites.</p>
</section>
<section id="differential-group-level-mcc-between-intact-and-lesion-region-among-each-tissue-type">
<h2>Differential group-level MCC between intact and lesion region among each tissue type<a class="headerlink" href="#differential-group-level-mcc-between-intact-and-lesion-region-among-each-tissue-type" title="Link to this heading"></a></h2>
<p>To further investigate how Parkinson’s disease affects MCC, we next examined the group-level MCC patterns associated with dopamine (HMDB0000073) and norepinephrine (HMDB0000216).
Specifically, we aimed to determine which tissue regions act as key senders or receivers of these signals, and how these communication relationships differ between the intact and lesion hemispheres.
This comparison provides insights into how dopaminergic and noradrenergic signaling networks are reorganized under pathological conditions.</p>
<p>Split adata into intact and lesion subsets</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_intact</span> <span class="o">=</span> <span class="n">adata</span><span class="p">[</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;lesion_shapes&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="kc">False</span><span class="p">,:]</span>
<span class="n">adata_lesion</span> <span class="o">=</span> <span class="n">adata</span><span class="p">[</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;lesion_shapes&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="kc">True</span><span class="p">,:]</span>
</pre></div>
</div>
</div>
</div>
<p>The original output of MetaChat provides MCC signals for each individual metabolite–sensor pair.
Here, we use <code class="docutils literal notranslate"><span class="pre">mc.tl.summary_communication</span></code> to summarize the overall MCC activity associated with specific metabolites (e.g., dopamine and norepinephrine), aggregating their communication strengths across all corresponding sensors.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sum_metabolites</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;HMDB0000216&#39;</span><span class="p">,</span> <span class="s1">&#39;HMDB0000073&#39;</span><span class="p">]</span>
<span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">summary_communication</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata_intact</span><span class="p">,</span>
                            <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                            <span class="n">sum_metabolites</span> <span class="o">=</span> <span class="n">sum_metabolites</span><span class="p">)</span>
<span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">summary_communication</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata_lesion</span><span class="p">,</span>
                            <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                            <span class="n">sum_metabolites</span> <span class="o">=</span> <span class="n">sum_metabolites</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>The function mc.tl.communication_group_spatial performs group-level MCC analysis by incorporating spatial permutation testing to evaluate whether a given signal shows significant interactions between different tissue types.
It outputs corresponding p-values that reflect the statistical significance of these spatial communication relationships.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">communication_group_spatial</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata_intact</span><span class="p">,</span>
                                  <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                                  <span class="n">group_name</span> <span class="o">=</span> <span class="s2">&quot;tissue_type&quot;</span><span class="p">,</span>
                                  <span class="n">sum_metabolites</span> <span class="o">=</span> <span class="n">sum_metabolites</span><span class="p">,</span>
                                  <span class="n">n_permutations</span> <span class="o">=</span> <span class="mi">100</span><span class="p">)</span>
<span class="n">mc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">communication_group_spatial</span><span class="p">(</span><span class="n">adata</span> <span class="o">=</span> <span class="n">adata_lesion</span><span class="p">,</span>
                                  <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                                  <span class="n">group_name</span> <span class="o">=</span> <span class="s2">&quot;tissue_type&quot;</span><span class="p">,</span>
                                  <span class="n">sum_metabolites</span> <span class="o">=</span> <span class="n">sum_metabolites</span><span class="p">,</span>
                                  <span class="n">n_permutations</span> <span class="o">=</span> <span class="mi">100</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>  Computing group-level MCC: 100%|██████████| 8100/8100 [00:24&lt;00:00, 326.55it/s]
  Computing group-level MCC: 100%|██████████| 8100/8100 [00:34&lt;00:00, 233.94it/s]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_group_communication_compare_hierarchy_diagram</span><span class="p">(</span><span class="n">adata_A</span> <span class="o">=</span> <span class="n">adata_intact</span><span class="p">,</span>
                                                         <span class="n">adata_B</span> <span class="o">=</span> <span class="n">adata_lesion</span><span class="p">,</span>
                                                         <span class="n">condition_name_A</span> <span class="o">=</span> <span class="s2">&quot;Intact&quot;</span><span class="p">,</span>
                                                         <span class="n">condition_name_B</span> <span class="o">=</span> <span class="s2">&quot;Lesion&quot;</span><span class="p">,</span>
                                                         <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                                                         <span class="n">group_name</span> <span class="o">=</span> <span class="s1">&#39;tissue_type&#39;</span><span class="p">,</span>
                                                         <span class="n">metabolite_name</span> <span class="o">=</span> <span class="s2">&quot;HMDB0000073&quot;</span><span class="p">,</span>
                                                         <span class="n">permutation_spatial</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                                                         <span class="n">p_value_cutoff</span> <span class="o">=</span> <span class="mf">0.05</span><span class="p">,</span>
                                                         <span class="n">node_sizes_limit</span> <span class="o">=</span> <span class="p">(</span><span class="mi">50</span><span class="p">,</span><span class="mi">300</span><span class="p">),</span>
                                                         <span class="n">edge_sizes_limit</span> <span class="o">=</span> <span class="p">(</span><span class="mf">0.5</span><span class="p">,</span><span class="mi">8</span><span class="p">),</span>
                                                         <span class="n">group_cmap</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
                                                         <span class="n">alpha</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span>
                                                         <span class="n">figsize</span> <span class="o">=</span> <span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">3</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;Axes: &gt;
</pre></div>
</div>
<img alt="../../_images/f833d52a68d823d9d6f70e0114a191bf261ecac89a3a7a6a9095cc6238b7804f.png" src="../../_images/f833d52a68d823d9d6f70e0114a191bf261ecac89a3a7a6a9095cc6238b7804f.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">plot_group_communication_compare_hierarchy_diagram</span><span class="p">(</span><span class="n">adata_A</span> <span class="o">=</span> <span class="n">adata_intact</span><span class="p">,</span>
                                                         <span class="n">adata_B</span> <span class="o">=</span> <span class="n">adata_lesion</span><span class="p">,</span>
                                                         <span class="n">condition_name_A</span> <span class="o">=</span> <span class="s2">&quot;Intact&quot;</span><span class="p">,</span>
                                                         <span class="n">condition_name_B</span> <span class="o">=</span> <span class="s2">&quot;Lesion&quot;</span><span class="p">,</span>
                                                         <span class="n">database_name</span> <span class="o">=</span> <span class="s1">&#39;MetaChatDB&#39;</span><span class="p">,</span>
                                                         <span class="n">group_name</span> <span class="o">=</span> <span class="s1">&#39;tissue_type&#39;</span><span class="p">,</span>
                                                         <span class="n">metabolite_name</span> <span class="o">=</span> <span class="s2">&quot;HMDB0000216&quot;</span><span class="p">,</span>
                                                         <span class="n">permutation_spatial</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                                                         <span class="n">p_value_cutoff</span> <span class="o">=</span> <span class="mf">0.05</span><span class="p">,</span>
                                                         <span class="n">node_sizes_limit</span> <span class="o">=</span> <span class="p">(</span><span class="mi">50</span><span class="p">,</span><span class="mi">300</span><span class="p">),</span>
                                                         <span class="n">edge_sizes_limit</span> <span class="o">=</span> <span class="p">(</span><span class="mf">0.5</span><span class="p">,</span><span class="mi">8</span><span class="p">),</span>
                                                         <span class="n">group_cmap</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
                                                         <span class="n">alpha</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span>
                                                         <span class="n">figsize</span> <span class="o">=</span> <span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mf">3.2</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;Axes: &gt;
</pre></div>
</div>
<img alt="../../_images/1f15960a84bdeb5c2962ba9c70481022d69ce31826b00d5b9df2920b4f21f10a.png" src="../../_images/1f15960a84bdeb5c2962ba9c70481022d69ce31826b00d5b9df2920b4f21f10a.png" />
</div>
</div>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="1.3_mouse_brain_parkinson_downstream.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">Downstream analysis on spatial multi-omics data of mouse brain with Parkinson’s disease</div>
              </div>
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="1.1_mouse_brain_parkinson_metachat.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">MetaChat analysis on spatial multi-omics data of mouse brain with Parkinson’s disease</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2025, Songhao Luo
            </div>
            Made with 
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            <div class="icons">
              <a class="muted-link fab fa-github" href="https://github.com/SonghaoLuo/metachat" aria-label="GitHub"></a>
              
            </div>
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Comparative analysis on spatial multi-omics data of mouse brain with Parkinson’s disease</a><ul>
<li><a class="reference internal" href="#import-metachat-results">Import MetaChat results</a></li>
<li><a class="reference internal" href="#clustering-through-mcc-signals">Clustering through MCC signals</a></li>
<li><a class="reference internal" href="#differential-mcc-among-each-tissue-type">Differential MCC among each tissue type</a></li>
<li><a class="reference internal" href="#differential-mcc-between-intact-and-lesion-region">Differential MCC between intact and lesion region</a></li>
<li><a class="reference internal" href="#differential-group-level-mcc-between-intact-and-lesion-region-among-each-tissue-type">Differential group-level MCC between intact and lesion region among each tissue type</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="../../_static/jquery.js?v=5d32c60e"></script>
    <script src="../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
    <script src="../../_static/documentation_options.js?v=282f96c0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/scripts/furo.js?v=46bd48cc"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    </body>
</html>